--- net/minecraft/src/BlockContainer.java
+++ net/minecraft/src/BlockContainer.java
@@ -1,23 +1,24 @@
 package net.minecraft.src;
 
 public abstract class BlockContainer extends Block {
-    protected BlockContainer(int var1, Material var2) {
-        super(var1, var2);
-        isBlockContainer[var1] = true;
-    }
-
-    protected BlockContainer(int var1, int var2, Material var3) {
-        super(var1, var2, var3);
-    }
-
-    public void onBlockAdded(World var1, int var2, int var3, int var4) {
-        super.onBlockAdded(var1, var2, var3, var4);
-        var1.setBlockTileEntity(var2, var3, var4, this.SetBlockEntity());
-    }
-
-    public void onBlockRemoval(World var1, int var2, int var3, int var4) {
-        super.onBlockRemoval(var1, var2, var3, var4);
-        var1.removeBlockTileEntity(var2, var3, var4);
+
+    protected BlockContainer(int i, Material material) {
+        super(i, material);
+        isBlockContainer[i] = true;
+    }
+
+    protected BlockContainer(int i, int j, Material material) {
+        super(i, j, material);
+    }
+
+    public void onBlockAdded(World world, int i, int j, int k) {
+        super.onBlockAdded(world, i, j, k);
+        world.setBlockTileEntity(i, j, k, SetBlockEntity());
+    }
+
+    public void onBlockRemoval(World world, int i, int j, int k) {
+        super.onBlockRemoval(world, i, j, k);
+        world.removeBlockTileEntity(i, j, k);
     }
 
     protected abstract TileEntity SetBlockEntity();
