--- net/minecraft/src/BlockGlass.java
+++ net/minecraft/src/BlockGlass.java
@@ -3,11 +3,12 @@
 import java.util.Random;
 
 public class BlockGlass extends BlockBreakable {
-    public BlockGlass(int var1, int var2, Material var3, boolean var4) {
-        super(var1, var2, var3, var4);
+
+    public BlockGlass(int i, int j, Material material, boolean flag) {
+        super(i, j, material, flag);
     }
 
-    public int quantityDropped(Random var1) {
+    public int quantityDropped(Random random) {
         return 0;
     }
 }
