--- net/minecraft/src/BlockGrass.java
+++ net/minecraft/src/BlockGrass.java
@@ -3,51 +3,51 @@
 import java.util.Random;
 
 public class BlockGrass extends Block {
-    protected BlockGrass(int var1) {
-        super(var1, Material.ground);
-        this.blockIndexInTexture = 3;
-        this.setTickOnLoad(true);
+
+    protected BlockGrass(int i) {
+        super(i, Material.ground);
+        blockIndexInTexture = 3;
+        setTickOnLoad(true);
     }
 
-    public int getBlockTexture(IBlockAccess var1, int var2, int var3, int var4, int var5) {
-        if(var5 == 1) {
+    public int getBlockTexture(IBlockAccess iblockaccess, int i, int j, int k, int l) {
+        if (l == 1) {
             return 0;
-        } else if(var5 == 0) {
+        }
+        if (l == 0) {
             return 2;
-        } else {
-            Material var6 = var1.getBlockMaterial(var2, var3 + 1, var4);
-            return var6 != Material.snow && var6 != Material.builtSnow ? 3 : 68;
-        }
-    }
-
-    public int colorMultiplier(IBlockAccess var1, int var2, int var3, int var4) {
-        var1.func_4075_a().func_4069_a(var2, var4, 1, 1);
-        double var5 = var1.func_4075_a().temperature[0];
-        double var7 = var1.func_4075_a().humidity[0];
-        return ColorizerGrass.func_4147_a(var5, var7);
-    }
-
-    public void updateTick(World var1, int var2, int var3, int var4, Random var5) {
-        if(!var1.multiplayerWorld) {
-            if(var1.getBlockLightValue(var2, var3 + 1, var4) < 4 && var1.getBlockMaterial(var2, var3 + 1, var4).getCanBlockGrass()) {
-                if(var5.nextInt(4) != 0) {
-                    return;
-                }
-
-                var1.setBlockWithNotify(var2, var3, var4, Block.dirt.blockID);
-            } else if(var1.getBlockLightValue(var2, var3 + 1, var4) >= 9) {
-                int var6 = var2 + var5.nextInt(3) - 1;
-                int var7 = var3 + var5.nextInt(5) - 3;
-                int var8 = var4 + var5.nextInt(3) - 1;
-                if(var1.getBlockId(var6, var7, var8) == Block.dirt.blockID && var1.getBlockLightValue(var6, var7 + 1, var8) >= 4 && !var1.getBlockMaterial(var6, var7 + 1, var8).getCanBlockGrass()) {
-                    var1.setBlockWithNotify(var6, var7, var8, Block.grass.blockID);
-                }
-            }
-
-        }
-    }
-
-    public int idDropped(int var1, Random var2) {
-        return Block.dirt.idDropped(0, var2);
+        }
+        Material material = iblockaccess.getBlockMaterial(i, j + 1, k);
+        return material != Material.snow && material != Material.builtSnow ? 3 : 68;
+    }
+
+    public int colorMultiplier(IBlockAccess iblockaccess, int i, int j, int k) {
+        iblockaccess.func_4075_a().func_4069_a(i, k, 1, 1);
+        double d = iblockaccess.func_4075_a().temperature[0];
+        double d1 = iblockaccess.func_4075_a().humidity[0];
+        return ColorizerGrass.func_4147_a(d, d1);
+    }
+
+    public void updateTick(World world, int i, int j, int k, Random random) {
+        if (world.multiplayerWorld) {
+            return;
+        }
+        if (world.getBlockLightValue(i, j + 1, k) < 4 && world.getBlockMaterial(i, j + 1, k).getCanBlockGrass()) {
+            if (random.nextInt(4) != 0) {
+                return;
+            }
+            world.setBlockWithNotify(i, j, k, Block.dirt.blockID);
+        } else if (world.getBlockLightValue(i, j + 1, k) >= 9) {
+            int l = (i + random.nextInt(3)) - 1;
+            int i1 = (j + random.nextInt(5)) - 3;
+            int j1 = (k + random.nextInt(3)) - 1;
+            if (world.getBlockId(l, i1, j1) == Block.dirt.blockID && world.getBlockLightValue(l, i1 + 1, j1) >= 4 && !world.getBlockMaterial(l, i1 + 1, j1).getCanBlockGrass()) {
+                world.setBlockWithNotify(l, i1, j1, Block.grass.blockID);
+            }
+        }
+    }
+
+    public int idDropped(int i, Random random) {
+        return Block.dirt.idDropped(0, random);
     }
 }
