--- net/minecraft/src/BlockLeavesBase.java
+++ net/minecraft/src/BlockLeavesBase.java
@@ -1,19 +1,24 @@
 package net.minecraft.src;
 
 public class BlockLeavesBase extends Block {
+
     protected boolean graphicsLevel;
 
-    protected BlockLeavesBase(int var1, int var2, Material var3, boolean var4) {
-        super(var1, var2, var3);
-        this.graphicsLevel = var4;
+    protected BlockLeavesBase(int i, int j, Material material, boolean flag) {
+        super(i, j, material);
+        graphicsLevel = flag;
     }
 
     public boolean isOpaqueCube() {
         return false;
     }
 
-    public boolean shouldSideBeRendered(IBlockAccess var1, int var2, int var3, int var4, int var5) {
-        int var6 = var1.getBlockId(var2, var3, var4);
-        return !this.graphicsLevel && var6 == this.blockID ? false : super.shouldSideBeRendered(var1, var2, var3, var4, var5);
+    public boolean shouldSideBeRendered(IBlockAccess iblockaccess, int i, int j, int k, int l) {
+        int i1 = iblockaccess.getBlockId(i, j, k);
+        if (!graphicsLevel && i1 == blockID) {
+            return false;
+        } else {
+            return super.shouldSideBeRendered(iblockaccess, i, j, k, l);
+        }
     }
 }
