--- net/minecraft/src/EntityOtherPlayerMP.java
+++ net/minecraft/src/EntityOtherPlayerMP.java
@@ -1,5 +1,8 @@
 package net.minecraft.src;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.world.player.PlayerLoadEvent;
+
 public class EntityOtherPlayerMP extends EntityPlayer {
     private int field_785_bg;
     private double field_784_bh;
@@ -15,8 +18,7 @@
         this.yOffset = 0.0F;
         this.stepHeight = 0.0F;
         if(var2 != null && var2.length() > 0) {
-            this.field_20047_bv = "http://www.minecraft.net/skin/" + var2 + ".png";
-            System.out.println("Loading texture " + this.field_20047_bv);
+            Ref.getEventManager().handleEvent(new PlayerLoadEvent(this));
         }
 
         this.noClip = true;
