--- net/minecraft/src/EntityArrow.java
+++ net/minecraft/src/EntityArrow.java
@@ -3,242 +3,246 @@
 import java.util.List;
 
 public class EntityArrow extends Entity {
-    private int xTile = -1;
-    private int yTile = -1;
-    private int zTile = -1;
-    private int inTile = 0;
-    private boolean inGround = false;
-    public int arrowShake = 0;
+
+    public int arrowShake;
     public EntityLiving field_682_g;
+    private int xTile;
+    private int yTile;
+    private int zTile;
+    private int inTile;
+    private boolean inGround;
     private int field_681_h;
-    private int field_680_i = 0;
-
-    public EntityArrow(World var1) {
-        super(var1);
-        this.setSize(0.5F, 0.5F);
-    }
-
-    public EntityArrow(World var1, double var2, double var4, double var6) {
-        super(var1);
-        this.setSize(0.5F, 0.5F);
-        this.setPosition(var2, var4, var6);
-        this.yOffset = 0.0F;
-    }
-
-    public EntityArrow(World var1, EntityLiving var2) {
-        super(var1);
-        this.field_682_g = var2;
-        this.setSize(0.5F, 0.5F);
-        this.setLocationAndAngles(var2.posX, var2.posY + (double)var2.func_373_s(), var2.posZ, var2.rotationYaw, var2.rotationPitch);
-        this.posX -= (double)(MathHelper.cos(this.rotationYaw / 180.0F * (float)Math.PI) * 0.16F);
-        this.posY -= (double)0.1F;
-        this.posZ -= (double)(MathHelper.sin(this.rotationYaw / 180.0F * (float)Math.PI) * 0.16F);
-        this.setPosition(this.posX, this.posY, this.posZ);
-        this.yOffset = 0.0F;
-        this.motionX = (double)(-MathHelper.sin(this.rotationYaw / 180.0F * (float)Math.PI) * MathHelper.cos(this.rotationPitch / 180.0F * (float)Math.PI));
-        this.motionZ = (double)(MathHelper.cos(this.rotationYaw / 180.0F * (float)Math.PI) * MathHelper.cos(this.rotationPitch / 180.0F * (float)Math.PI));
-        this.motionY = (double)(-MathHelper.sin(this.rotationPitch / 180.0F * (float)Math.PI));
-        this.setArrowHeading(this.motionX, this.motionY, this.motionZ, 1.5F, 1.0F);
-    }
-
-    public void setArrowHeading(double var1, double var3, double var5, float var7, float var8) {
-        float var9 = MathHelper.sqrt_double(var1 * var1 + var3 * var3 + var5 * var5);
-        var1 /= (double)var9;
-        var3 /= (double)var9;
-        var5 /= (double)var9;
-        var1 += this.rand.nextGaussian() * (double)0.0075F * (double)var8;
-        var3 += this.rand.nextGaussian() * (double)0.0075F * (double)var8;
-        var5 += this.rand.nextGaussian() * (double)0.0075F * (double)var8;
-        var1 *= (double)var7;
-        var3 *= (double)var7;
-        var5 *= (double)var7;
-        this.motionX = var1;
-        this.motionY = var3;
-        this.motionZ = var5;
-        float var10 = MathHelper.sqrt_double(var1 * var1 + var5 * var5);
-        this.prevRotationYaw = this.rotationYaw = (float)(Math.atan2(var1, var5) * 180.0D / (double)((float)Math.PI));
-        this.prevRotationPitch = this.rotationPitch = (float)(Math.atan2(var3, (double)var10) * 180.0D / (double)((float)Math.PI));
-        this.field_681_h = 0;
-    }
-
-    public void setVelocity(double var1, double var3, double var5) {
-        this.motionX = var1;
-        this.motionY = var3;
-        this.motionZ = var5;
-        if(this.prevRotationPitch == 0.0F && this.prevRotationYaw == 0.0F) {
-            float var7 = MathHelper.sqrt_double(var1 * var1 + var5 * var5);
-            this.prevRotationYaw = this.rotationYaw = (float)(Math.atan2(var1, var5) * 180.0D / (double)((float)Math.PI));
-            this.prevRotationPitch = this.rotationPitch = (float)(Math.atan2(var3, (double)var7) * 180.0D / (double)((float)Math.PI));
+    private int field_680_i;
+
+    public EntityArrow(World world) {
+        super(world);
+        xTile = -1;
+        yTile = -1;
+        zTile = -1;
+        inTile = 0;
+        inGround = false;
+        arrowShake = 0;
+        field_680_i = 0;
+        setSize(0.5F, 0.5F);
+    }
+
+    public EntityArrow(World world, double d, double d1, double d2) {
+        super(world);
+        xTile = -1;
+        yTile = -1;
+        zTile = -1;
+        inTile = 0;
+        inGround = false;
+        arrowShake = 0;
+        field_680_i = 0;
+        setSize(0.5F, 0.5F);
+        setPosition(d, d1, d2);
+        yOffset = 0.0F;
+    }
+    public EntityArrow(World world, EntityLiving entityliving) {
+        super(world);
+        xTile = -1;
+        yTile = -1;
+        zTile = -1;
+        inTile = 0;
+        inGround = false;
+        arrowShake = 0;
+        field_680_i = 0;
+        field_682_g = entityliving;
+        setSize(0.5F, 0.5F);
+        setLocationAndAngles(entityliving.posX, entityliving.posY + (double) entityliving.func_373_s(), entityliving.posZ, entityliving.rotationYaw, entityliving.rotationPitch);
+        posX -= MathHelper.cos((rotationYaw / 180F) * 3.141593F) * 0.16F;
+        posY -= 0.10000000149011612D;
+        posZ -= MathHelper.sin((rotationYaw / 180F) * 3.141593F) * 0.16F;
+        setPosition(posX, posY, posZ);
+        yOffset = 0.0F;
+        motionX = -MathHelper.sin((rotationYaw / 180F) * 3.141593F) * MathHelper.cos((rotationPitch / 180F) * 3.141593F);
+        motionZ = MathHelper.cos((rotationYaw / 180F) * 3.141593F) * MathHelper.cos((rotationPitch / 180F) * 3.141593F);
+        motionY = -MathHelper.sin((rotationPitch / 180F) * 3.141593F);
+        setArrowHeading(motionX, motionY, motionZ, 1.5F, 1.0F);
+    }
+
+    public void setArrowHeading(double d, double d1, double d2, float f,
+                                float f1) {
+        float f2 = MathHelper.sqrt_double(d * d + d1 * d1 + d2 * d2);
+        d /= f2;
+        d1 /= f2;
+        d2 /= f2;
+        d += rand.nextGaussian() * 0.0074999998323619366D * (double) f1;
+        d1 += rand.nextGaussian() * 0.0074999998323619366D * (double) f1;
+        d2 += rand.nextGaussian() * 0.0074999998323619366D * (double) f1;
+        d *= f;
+        d1 *= f;
+        d2 *= f;
+        motionX = d;
+        motionY = d1;
+        motionZ = d2;
+        float f3 = MathHelper.sqrt_double(d * d + d2 * d2);
+        prevRotationYaw = rotationYaw = (float) ((Math.atan2(d, d2) * 180D) / 3.1415927410125732D);
+        prevRotationPitch = rotationPitch = (float) ((Math.atan2(d1, f3) * 180D) / 3.1415927410125732D);
+        field_681_h = 0;
+    }
+
+    public void setVelocity(double d, double d1, double d2) {
+        motionX = d;
+        motionY = d1;
+        motionZ = d2;
+        if (prevRotationPitch == 0.0F && prevRotationYaw == 0.0F) {
+            float f = MathHelper.sqrt_double(d * d + d2 * d2);
+            prevRotationYaw = rotationYaw = (float) ((Math.atan2(d, d2) * 180D) / 3.1415927410125732D);
+            prevRotationPitch = rotationPitch = (float) ((Math.atan2(d1, f) * 180D) / 3.1415927410125732D);
         }
-
     }
 
     public void onUpdate() {
         super.onUpdate();
-        if(this.prevRotationPitch == 0.0F && this.prevRotationYaw == 0.0F) {
-            float var1 = MathHelper.sqrt_double(this.motionX * this.motionX + this.motionZ * this.motionZ);
-            this.prevRotationYaw = this.rotationYaw = (float)(Math.atan2(this.motionX, this.motionZ) * 180.0D / (double)((float)Math.PI));
-            this.prevRotationPitch = this.rotationPitch = (float)(Math.atan2(this.motionY, (double)var1) * 180.0D / (double)((float)Math.PI));
-        }
-
-        if(this.arrowShake > 0) {
-            --this.arrowShake;
-        }
-
-        if(this.inGround) {
-            int var15 = this.worldObj.getBlockId(this.xTile, this.yTile, this.zTile);
-            if(var15 == this.inTile) {
-                ++this.field_681_h;
-                if(this.field_681_h == 1200) {
-                    this.setEntityDead();
+        if (prevRotationPitch == 0.0F && prevRotationYaw == 0.0F) {
+            float f = MathHelper.sqrt_double(motionX * motionX + motionZ * motionZ);
+            prevRotationYaw = rotationYaw = (float) ((Math.atan2(motionX, motionZ) * 180D) / 3.1415927410125732D);
+            prevRotationPitch = rotationPitch = (float) ((Math.atan2(motionY, f) * 180D) / 3.1415927410125732D);
+        }
+        if (arrowShake > 0) {
+            arrowShake--;
+        }
+        if (inGround) {
+            int i = worldObj.getBlockId(xTile, yTile, zTile);
+            if (i != inTile) {
+                inGround = false;
+                motionX *= rand.nextFloat() * 0.2F;
+                motionY *= rand.nextFloat() * 0.2F;
+                motionZ *= rand.nextFloat() * 0.2F;
+                field_681_h = 0;
+                field_680_i = 0;
+            } else {
+                field_681_h++;
+                if (field_681_h == 1200) {
+                    setEntityDead();
                 }
-
                 return;
             }
-
-            this.inGround = false;
-            this.motionX *= (double)(this.rand.nextFloat() * 0.2F);
-            this.motionY *= (double)(this.rand.nextFloat() * 0.2F);
-            this.motionZ *= (double)(this.rand.nextFloat() * 0.2F);
-            this.field_681_h = 0;
-            this.field_680_i = 0;
         } else {
-            ++this.field_680_i;
-        }
-
-        Vec3D var16 = Vec3D.createVector(this.posX, this.posY, this.posZ);
-        Vec3D var2 = Vec3D.createVector(this.posX + this.motionX, this.posY + this.motionY, this.posZ + this.motionZ);
-        MovingObjectPosition var3 = this.worldObj.rayTraceBlocks(var16, var2);
-        var16 = Vec3D.createVector(this.posX, this.posY, this.posZ);
-        var2 = Vec3D.createVector(this.posX + this.motionX, this.posY + this.motionY, this.posZ + this.motionZ);
-        if(var3 != null) {
-            var2 = Vec3D.createVector(var3.hitVec.xCoord, var3.hitVec.yCoord, var3.hitVec.zCoord);
-        }
-
-        Entity var4 = null;
-        List var5 = this.worldObj.getEntitiesWithinAABBExcludingEntity(this, this.boundingBox.addCoord(this.motionX, this.motionY, this.motionZ).expand(1.0D, 1.0D, 1.0D));
-        double var6 = 0.0D;
-
-        float var10;
-        for(int var8 = 0; var8 < var5.size(); ++var8) {
-            Entity var9 = (Entity)var5.get(var8);
-            if(var9.canBeCollidedWith() && (var9 != this.field_682_g || this.field_680_i >= 5)) {
-                var10 = 0.3F;
-                AxisAlignedBB var11 = var9.boundingBox.expand((double)var10, (double)var10, (double)var10);
-                MovingObjectPosition var12 = var11.func_1169_a(var16, var2);
-                if(var12 != null) {
-                    double var13 = var16.distanceTo(var12.hitVec);
-                    if(var13 < var6 || var6 == 0.0D) {
-                        var4 = var9;
-                        var6 = var13;
-                    }
-                }
-            }
-        }
-
-        if(var4 != null) {
-            var3 = new MovingObjectPosition(var4);
-        }
-
-        float var17;
-        if(var3 != null) {
-            if(var3.entityHit != null) {
-                if(var3.entityHit.attackEntityFrom(this.field_682_g, 4)) {
-                    this.worldObj.playSoundAtEntity(this, "random.drr", 1.0F, 1.2F / (this.rand.nextFloat() * 0.2F + 0.9F));
-                    this.setEntityDead();
+            field_680_i++;
+        }
+        Vec3D vec3d = Vec3D.createVector(posX, posY, posZ);
+        Vec3D vec3d1 = Vec3D.createVector(posX + motionX, posY + motionY, posZ + motionZ);
+        MovingObjectPosition movingobjectposition = worldObj.rayTraceBlocks(vec3d, vec3d1);
+        vec3d = Vec3D.createVector(posX, posY, posZ);
+        vec3d1 = Vec3D.createVector(posX + motionX, posY + motionY, posZ + motionZ);
+        if (movingobjectposition != null) {
+            vec3d1 = Vec3D.createVector(movingobjectposition.hitVec.xCoord, movingobjectposition.hitVec.yCoord, movingobjectposition.hitVec.zCoord);
+        }
+        Entity entity = null;
+        List list = worldObj.getEntitiesWithinAABBExcludingEntity(this, boundingBox.addCoord(motionX, motionY, motionZ).expand(1.0D, 1.0D, 1.0D));
+        double d = 0.0D;
+        for (int j = 0; j < list.size(); j++) {
+            Entity entity1 = (Entity) list.get(j);
+            if (!entity1.canBeCollidedWith() || entity1 == field_682_g && field_680_i < 5) {
+                continue;
+            }
+            float f4 = 0.3F;
+            AxisAlignedBB axisalignedbb = entity1.boundingBox.expand(f4, f4, f4);
+            MovingObjectPosition movingobjectposition1 = axisalignedbb.func_1169_a(vec3d, vec3d1);
+            if (movingobjectposition1 == null) {
+                continue;
+            }
+            double d1 = vec3d.distanceTo(movingobjectposition1.hitVec);
+            if (d1 < d || d == 0.0D) {
+                entity = entity1;
+                d = d1;
+            }
+        }
+
+        if (entity != null) {
+            movingobjectposition = new MovingObjectPosition(entity);
+        }
+        if (movingobjectposition != null) {
+            if (movingobjectposition.entityHit != null) {
+                if (movingobjectposition.entityHit.attackEntityFrom(field_682_g, 4)) {
+                    worldObj.playSoundAtEntity(this, "random.drr", 1.0F, 1.2F / (rand.nextFloat() * 0.2F + 0.9F));
+                    setEntityDead();
                 } else {
-                    this.motionX *= (double)-0.1F;
-                    this.motionY *= (double)-0.1F;
-                    this.motionZ *= (double)-0.1F;
-                    this.rotationYaw += 180.0F;
-                    this.prevRotationYaw += 180.0F;
-                    this.field_680_i = 0;
+                    motionX *= -0.10000000149011612D;
+                    motionY *= -0.10000000149011612D;
+                    motionZ *= -0.10000000149011612D;
+                    rotationYaw += 180F;
+                    prevRotationYaw += 180F;
+                    field_680_i = 0;
                 }
             } else {
-                this.xTile = var3.blockX;
-                this.yTile = var3.blockY;
-                this.zTile = var3.blockZ;
-                this.inTile = this.worldObj.getBlockId(this.xTile, this.yTile, this.zTile);
-                this.motionX = (double)((float)(var3.hitVec.xCoord - this.posX));
-                this.motionY = (double)((float)(var3.hitVec.yCoord - this.posY));
-                this.motionZ = (double)((float)(var3.hitVec.zCoord - this.posZ));
-                var17 = MathHelper.sqrt_double(this.motionX * this.motionX + this.motionY * this.motionY + this.motionZ * this.motionZ);
-                this.posX -= this.motionX / (double)var17 * (double)0.05F;
-                this.posY -= this.motionY / (double)var17 * (double)0.05F;
-                this.posZ -= this.motionZ / (double)var17 * (double)0.05F;
-                this.worldObj.playSoundAtEntity(this, "random.drr", 1.0F, 1.2F / (this.rand.nextFloat() * 0.2F + 0.9F));
-                this.inGround = true;
-                this.arrowShake = 7;
-            }
-        }
-
-        this.posX += this.motionX;
-        this.posY += this.motionY;
-        this.posZ += this.motionZ;
-        var17 = MathHelper.sqrt_double(this.motionX * this.motionX + this.motionZ * this.motionZ);
-        this.rotationYaw = (float)(Math.atan2(this.motionX, this.motionZ) * 180.0D / (double)((float)Math.PI));
-
-        for(this.rotationPitch = (float)(Math.atan2(this.motionY, (double)var17) * 180.0D / (double)((float)Math.PI)); this.rotationPitch - this.prevRotationPitch < -180.0F; this.prevRotationPitch -= 360.0F) {
-        }
-
-        while(this.rotationPitch - this.prevRotationPitch >= 180.0F) {
-            this.prevRotationPitch += 360.0F;
-        }
-
-        while(this.rotationYaw - this.prevRotationYaw < -180.0F) {
-            this.prevRotationYaw -= 360.0F;
-        }
-
-        while(this.rotationYaw - this.prevRotationYaw >= 180.0F) {
-            this.prevRotationYaw += 360.0F;
-        }
-
-        this.rotationPitch = this.prevRotationPitch + (this.rotationPitch - this.prevRotationPitch) * 0.2F;
-        this.rotationYaw = this.prevRotationYaw + (this.rotationYaw - this.prevRotationYaw) * 0.2F;
-        float var18 = 0.99F;
-        var10 = 0.03F;
-        if(this.handleWaterMovement()) {
-            for(int var19 = 0; var19 < 4; ++var19) {
-                float var20 = 0.25F;
-                this.worldObj.spawnParticle("bubble", this.posX - this.motionX * (double)var20, this.posY - this.motionY * (double)var20, this.posZ - this.motionZ * (double)var20, this.motionX, this.motionY, this.motionZ);
-            }
-
-            var18 = 0.8F;
-        }
-
-        this.motionX *= (double)var18;
-        this.motionY *= (double)var18;
-        this.motionZ *= (double)var18;
-        this.motionY -= (double)var10;
-        this.setPosition(this.posX, this.posY, this.posZ);
-    }
-
-    public void writeEntityToNBT(NBTTagCompound var1) {
-        var1.setShort("xTile", (short)this.xTile);
-        var1.setShort("yTile", (short)this.yTile);
-        var1.setShort("zTile", (short)this.zTile);
-        var1.setByte("inTile", (byte)this.inTile);
-        var1.setByte("shake", (byte)this.arrowShake);
-        var1.setByte("inGround", (byte)(this.inGround ? 1 : 0));
-    }
-
-    public void readEntityFromNBT(NBTTagCompound var1) {
-        this.xTile = var1.getShort("xTile");
-        this.yTile = var1.getShort("yTile");
-        this.zTile = var1.getShort("zTile");
-        this.inTile = var1.getByte("inTile") & 255;
-        this.arrowShake = var1.getByte("shake") & 255;
-        this.inGround = var1.getByte("inGround") == 1;
-    }
-
-    public void onCollideWithPlayer(EntityPlayer var1) {
-        if(!this.worldObj.multiplayerWorld) {
-            if(this.inGround && this.field_682_g == var1 && this.arrowShake <= 0 && var1.inventory.addItemStackToInventory(new ItemStack(Item.arrow.shiftedIndex, 1))) {
-                this.worldObj.playSoundAtEntity(this, "random.pop", 0.2F, ((this.rand.nextFloat() - this.rand.nextFloat()) * 0.7F + 1.0F) * 2.0F);
-                var1.onItemPickup(this, 1);
-                this.setEntityDead();
-            }
-
+                xTile = movingobjectposition.blockX;
+                yTile = movingobjectposition.blockY;
+                zTile = movingobjectposition.blockZ;
+                inTile = worldObj.getBlockId(xTile, yTile, zTile);
+                motionX = (float) (movingobjectposition.hitVec.xCoord - posX);
+                motionY = (float) (movingobjectposition.hitVec.yCoord - posY);
+                motionZ = (float) (movingobjectposition.hitVec.zCoord - posZ);
+                float f1 = MathHelper.sqrt_double(motionX * motionX + motionY * motionY + motionZ * motionZ);
+                posX -= (motionX / (double) f1) * 0.05000000074505806D;
+                posY -= (motionY / (double) f1) * 0.05000000074505806D;
+                posZ -= (motionZ / (double) f1) * 0.05000000074505806D;
+                worldObj.playSoundAtEntity(this, "random.drr", 1.0F, 1.2F / (rand.nextFloat() * 0.2F + 0.9F));
+                inGround = true;
+                arrowShake = 7;
+            }
+        }
+        posX += motionX;
+        posY += motionY;
+        posZ += motionZ;
+        float f2 = MathHelper.sqrt_double(motionX * motionX + motionZ * motionZ);
+        rotationYaw = (float) ((Math.atan2(motionX, motionZ) * 180D) / 3.1415927410125732D);
+        for (rotationPitch = (float) ((Math.atan2(motionY, f2) * 180D) / 3.1415927410125732D); rotationPitch - prevRotationPitch < -180F; prevRotationPitch -= 360F) {
+        }
+        for (; rotationPitch - prevRotationPitch >= 180F; prevRotationPitch += 360F) {
+        }
+        for (; rotationYaw - prevRotationYaw < -180F; prevRotationYaw -= 360F) {
+        }
+        for (; rotationYaw - prevRotationYaw >= 180F; prevRotationYaw += 360F) {
+        }
+        rotationPitch = prevRotationPitch + (rotationPitch - prevRotationPitch) * 0.2F;
+        rotationYaw = prevRotationYaw + (rotationYaw - prevRotationYaw) * 0.2F;
+        float f3 = 0.99F;
+        float f5 = 0.03F;
+        if (handleWaterMovement()) {
+            for (int k = 0; k < 4; k++) {
+                float f6 = 0.25F;
+                worldObj.spawnParticle("bubble", posX - motionX * (double) f6, posY - motionY * (double) f6, posZ - motionZ * (double) f6, motionX, motionY, motionZ);
+            }
+
+            f3 = 0.8F;
+        }
+        motionX *= f3;
+        motionY *= f3;
+        motionZ *= f3;
+        motionY -= f5;
+        setPosition(posX, posY, posZ);
+    }
+
+    public void writeEntityToNBT(NBTTagCompound nbttagcompound) {
+        nbttagcompound.setShort("xTile", (short) xTile);
+        nbttagcompound.setShort("yTile", (short) yTile);
+        nbttagcompound.setShort("zTile", (short) zTile);
+        nbttagcompound.setByte("inTile", (byte) inTile);
+        nbttagcompound.setByte("shake", (byte) arrowShake);
+        nbttagcompound.setByte("inGround", (byte) (inGround ? 1 : 0));
+    }
+
+    public void readEntityFromNBT(NBTTagCompound nbttagcompound) {
+        xTile = nbttagcompound.getShort("xTile");
+        yTile = nbttagcompound.getShort("yTile");
+        zTile = nbttagcompound.getShort("zTile");
+        inTile = nbttagcompound.getByte("inTile") & 0xff;
+        arrowShake = nbttagcompound.getByte("shake") & 0xff;
+        inGround = nbttagcompound.getByte("inGround") == 1;
+    }
+
+    public void onCollideWithPlayer(EntityPlayer entityplayer) {
+        if (worldObj.multiplayerWorld) {
+            return;
+        }
+        if (inGround && field_682_g == entityplayer && arrowShake <= 0 && entityplayer.inventory.addItemStackToInventory(new ItemStack(Item.arrow.shiftedIndex, 1))) {
+            worldObj.playSoundAtEntity(this, "random.pop", 0.2F, ((rand.nextFloat() - rand.nextFloat()) * 0.7F + 1.0F) * 2.0F);
+            entityplayer.onItemPickup(this, 1);
+            setEntityDead();
         }
     }
 
