--- net/minecraft/src/EntityClientPlayerMP.java
+++ net/minecraft/src/EntityClientPlayerMP.java
@@ -1,5 +1,7 @@
 package net.minecraft.src;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.chat.ChatSendEvent;
 import net.minecraft.client.Minecraft;
 
 public class EntityClientPlayerMP extends EntityPlayerSP {
@@ -114,6 +116,13 @@
     }
 
     public void sendChatMessage(String var1) {
+        ChatSendEvent event = new ChatSendEvent(var1);
+        Ref.getEventManager().handleEvent(event);
+
+        if (event.isCancelled()) {
+            return;
+        }
+
         this.field_797_bg.addToSendQueue(new Packet3Chat(var1));
     }
 
