--- net/minecraft/src/EntityPlayerSP.java
+++ net/minecraft/src/EntityPlayerSP.java
@@ -1,5 +1,7 @@
 package net.minecraft.src;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.world.player.PlayerLoadEvent;
 import net.minecraft.client.Minecraft;
 
 public class EntityPlayerSP extends EntityPlayer {
@@ -15,11 +17,9 @@
         this.mc = var1;
         this.dimension = var4;
         if(var3 != null && var3.playerName != null && var3.playerName.length() > 0) {
-            this.field_20047_bv = "http://www.minecraft.net/skin/" + var3.playerName + ".png";
-            System.out.println("Loading texture " + this.field_20047_bv);
+            this.field_771_i = var3.playerName;
+            Ref.getEventManager().handleEvent(new PlayerLoadEvent(this));
         }
-
-        this.field_771_i = var3.playerName;
     }
 
     public void updatePlayerActionState() {
