--- net/minecraft/src/EntityPlayer.java
+++ net/minecraft/src/EntityPlayer.java
@@ -2,6 +2,9 @@
 
 import java.util.List;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.world.player.OnLivingUpdateEvent;
+
 public abstract class EntityPlayer extends EntityLiving {
     public InventoryPlayer inventory = new InventoryPlayer(this);
     public CraftingInventoryCB field_20069_g;
@@ -118,6 +121,9 @@
     }
 
     public void onLivingUpdate() {
+        OnLivingUpdateEvent event = new OnLivingUpdateEvent(this);
+        Ref.getEventManager().handleEvent(event);
+
         if(this.worldObj.difficultySetting == 0 && this.health < 20 && this.ticksExisted % 20 * 12 == 0) {
             this.heal(1);
         }
