--- net/minecraft/src/EntityRenderer.java
+++ net/minecraft/src/EntityRenderer.java
@@ -12,6 +12,14 @@
 import org.lwjgl.opengl.NVFogDistance;
 import org.lwjgl.util.glu.GLU;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.gui.PortalOverlayDrawEvent;
+import cc.noxiuam.titanic.event.impl.mouse.PlayerLookInputEvent;
+import cc.noxiuam.titanic.event.impl.perspective.CameraChangeEvent;
+import cc.noxiuam.titanic.event.impl.perspective.ViewBobbingSetupEvent;
+import cc.noxiuam.titanic.event.impl.world.fov.FOVUpdateEvent;
+import cc.noxiuam.titanic.event.impl.world.fov.PreFOVUpdateEvent;
+
 public class EntityRenderer {
     private Minecraft mc;
     private float farPlaneDistance = 0.0F;
@@ -103,6 +111,7 @@
         }
     }
 
+    // fov
     private float func_914_d(float var1) {
         EntityPlayerSP var2 = this.mc.thePlayer;
         float var3 = 70.0F;
@@ -110,12 +119,19 @@
             var3 = 60.0F;
         }
 
+        PreFOVUpdateEvent preFovUpdateEvent = new PreFOVUpdateEvent(var3);
+        Ref.getEventManager().handleEvent(preFovUpdateEvent);
+        var3 = preFovUpdateEvent.getFov();
+
         if(var2.health <= 0) {
             float var4 = (float)var2.deathTime + var1;
             var3 /= (1.0F - 500.0F / (var4 + 500.0F)) * 2.0F + 1.0F;
         }
 
-        return var3;
+
+        FOVUpdateEvent event = new FOVUpdateEvent(var1, var3);
+        Ref.getEventManager().handleEvent(event);
+        return event.getFov();
     }
 
     private void hurtCameraEffect(float var1) {
@@ -138,6 +154,13 @@
     }
 
     private void setupViewBobbing(float var1) {
+        ViewBobbingSetupEvent event = new ViewBobbingSetupEvent(var1);
+        Ref.getEventManager().handleEvent(event);
+
+        if (event.isCancelled()) {
+            return;
+        }
+
         if(!this.mc.gameSettings.thirdPersonView) {
             EntityPlayerSP var2 = this.mc.thePlayer;
             float var3 = var2.distanceWalkedModified - var2.prevDistanceWalkedModified;
@@ -165,6 +188,13 @@
                 var9 += 2.0D;
             }
 
+            CameraChangeEvent event = new CameraChangeEvent(var12);
+            Ref.getEventManager().handleEvent(event);
+
+            if (event.isCancelled()) {
+                var12 = event.getPitch();
+            }
+
             double var13 = (double)(-MathHelper.sin(var11 / 180.0F * (float)Math.PI) * MathHelper.cos(var12 / 180.0F * (float)Math.PI)) * var9;
             double var15 = (double)(MathHelper.cos(var11 / 180.0F * (float)Math.PI) * MathHelper.cos(var12 / 180.0F * (float)Math.PI)) * var9;
             double var17 = (double)(-MathHelper.sin(var12 / 180.0F * (float)Math.PI)) * var9;
@@ -224,6 +254,14 @@
         }
 
         float var4 = this.mc.thePlayer.prevTimeInPortal + (this.mc.thePlayer.timeInPortal - this.mc.thePlayer.prevTimeInPortal) * var1;
+
+        PortalOverlayDrawEvent portalOverlayDrawEvent = new PortalOverlayDrawEvent(mc.thePlayer.prevTimeInPortal, mc.thePlayer.timeInPortal, var4);
+        Ref.getEventManager().handleEvent(portalOverlayDrawEvent);
+
+        if (portalOverlayDrawEvent.isCancelled()) {
+            var4 = portalOverlayDrawEvent.prevTimeInPortal + (portalOverlayDrawEvent.timeInPortal - portalOverlayDrawEvent.prevTimeInPortal) * var4;
+        }
+
         if(var4 > 0.0F) {
             float var5 = 5.0F / (var4 * var4 + 5.0F) - var4 * 0.04F;
             var5 *= var5;
@@ -278,12 +316,16 @@
             float var3 = var2 * var2 * var2 * 8.0F;
             float var4 = (float)this.mc.mouseHelper.field_1114_a * var3;
             float var5 = (float)this.mc.mouseHelper.field_1113_b * var3;
-            byte var6 = 1;
             if(this.mc.gameSettings.invertMouse) {
-                var6 = -1;
-            }
-
-            this.mc.thePlayer.func_346_d(var4, var5 * (float)var6);
+                var5 = -var5;
+            }
+
+            PlayerLookInputEvent event = new PlayerLookInputEvent(var4, var5, var1);
+            Ref.getEventManager().handleEvent(event);
+
+            if (!event.isCancelled()) {
+                this.mc.thePlayer.func_346_d(event.getDx(), event.getDy());
+            }
         }
 
         if(!this.mc.field_6307_v) {
