--- net/minecraft/src/EntityZombie.java
+++ net/minecraft/src/EntityZombie.java
@@ -1,21 +1,21 @@
 package net.minecraft.src;
 
 public class EntityZombie extends EntityMobs {
-    public EntityZombie(World var1) {
-        super(var1);
-        this.texture = "/mob/zombie.png";
-        this.moveSpeed = 0.5F;
-        this.attackStrength = 5;
+
+    public EntityZombie(World world) {
+        super(world);
+        texture = "/mob/zombie.png";
+        moveSpeed = 0.5F;
+        attackStrength = 5;
     }
 
     public void onLivingUpdate() {
-        if(this.worldObj.isDaytime()) {
-            float var1 = this.getEntityBrightness(1.0F);
-            if(var1 > 0.5F && this.worldObj.canBlockSeeTheSky(MathHelper.floor_double(this.posX), MathHelper.floor_double(this.posY), MathHelper.floor_double(this.posZ)) && this.rand.nextFloat() * 30.0F < (var1 - 0.4F) * 2.0F) {
-                this.fire = 300;
+        if (worldObj.isDaytime()) {
+            float f = getEntityBrightness(1.0F);
+            if (f > 0.5F && worldObj.canBlockSeeTheSky(MathHelper.floor_double(posX), MathHelper.floor_double(posY), MathHelper.floor_double(posZ)) && rand.nextFloat() * 30F < (f - 0.4F) * 2.0F) {
+                fire = 300;
             }
         }
-
         super.onLivingUpdate();
     }
 
