--- net/minecraft/src/EntityPig.java
+++ net/minecraft/src/EntityPig.java
@@ -1,23 +1,25 @@
 package net.minecraft.src;
 
 public class EntityPig extends EntityAnimals {
-    public boolean rideable = false;
-
-    public EntityPig(World var1) {
-        super(var1);
-        this.texture = "/mob/pig.png";
-        this.setSize(0.9F, 0.9F);
-        this.rideable = false;
-    }
-
-    public void writeEntityToNBT(NBTTagCompound var1) {
-        super.writeEntityToNBT(var1);
-        var1.setBoolean("Saddle", this.rideable);
-    }
-
-    public void readEntityFromNBT(NBTTagCompound var1) {
-        super.readEntityFromNBT(var1);
-        this.rideable = var1.getBoolean("Saddle");
+
+    public boolean rideable;
+
+    public EntityPig(World world) {
+        super(world);
+        rideable = false;
+        texture = "/mob/pig.png";
+        setSize(0.9F, 0.9F);
+        rideable = false;
+    }
+
+    public void writeEntityToNBT(NBTTagCompound nbttagcompound) {
+        super.writeEntityToNBT(nbttagcompound);
+        nbttagcompound.setBoolean("Saddle", rideable);
+    }
+
+    public void readEntityFromNBT(NBTTagCompound nbttagcompound) {
+        super.readEntityFromNBT(nbttagcompound);
+        rideable = nbttagcompound.getBoolean("Saddle");
     }
 
     protected String getLivingSound() {
@@ -32,9 +34,9 @@
         return "mob.pigdeath";
     }
 
-    public boolean interact(EntityPlayer var1) {
-        if(this.rideable) {
-            var1.mountEntity(this);
+    public boolean interact(EntityPlayer entityplayer) {
+        if (rideable) {
+            entityplayer.mountEntity(this);
             return true;
         } else {
             return false;
