--- net/minecraft/src/EntitySmokeFX.java
+++ net/minecraft/src/EntitySmokeFX.java
@@ -1,63 +1,60 @@
 package net.minecraft.src;
 
 public class EntitySmokeFX extends EntityFX {
+
     float field_671_a;
 
-    public EntitySmokeFX(World var1, double var2, double var4, double var6) {
-        this(var1, var2, var4, var6, 1.0F);
-    }
-
-    public EntitySmokeFX(World var1, double var2, double var4, double var6, float var8) {
-        super(var1, var2, var4, var6, 0.0D, 0.0D, 0.0D);
-        this.motionX *= (double)0.1F;
-        this.motionY *= (double)0.1F;
-        this.motionZ *= (double)0.1F;
-        this.particleRed = this.particleBlue = this.particleGreen = (float)(Math.random() * (double)0.3F);
-        this.field_665_g *= 12.0F / 16.0F;
-        this.field_665_g *= var8;
-        this.field_671_a = this.field_665_g;
-        this.field_666_f = (int)(8.0D / (Math.random() * 0.8D + 0.2D));
-        this.field_666_f = (int)((float)this.field_666_f * var8);
-        this.noClip = false;
-    }
-
-    public void func_406_a(Tessellator var1, float var2, float var3, float var4, float var5, float var6, float var7) {
-        float var8 = ((float)this.e + var2) / (float)this.field_666_f * 32.0F;
-        if(var8 < 0.0F) {
-            var8 = 0.0F;
-        }
-
-        if(var8 > 1.0F) {
-            var8 = 1.0F;
-        }
-
-        this.field_665_g = this.field_671_a * var8;
-        super.func_406_a(var1, var2, var3, var4, var5, var6, var7);
+    public EntitySmokeFX(World world, double d, double d1, double d2) {
+        this(world, d, d1, d2, 1.0F);
+    }
+
+    public EntitySmokeFX(World world, double d, double d1, double d2,
+                         float f) {
+        super(world, d, d1, d2, 0.0D, 0.0D, 0.0D);
+        motionX *= 0.10000000149011612D;
+        motionY *= 0.10000000149011612D;
+        motionZ *= 0.10000000149011612D;
+        particleRed = particleBlue = particleGreen = (float) (Math.random() * 0.30000001192092896D);
+        field_665_g *= 0.75F;
+        field_665_g *= f;
+        field_671_a = field_665_g;
+        field_666_f = (int) (8D / (Math.random() * 0.80000000000000004D + 0.20000000000000001D));
+        field_666_f *= f;
+        noClip = false;
+    }
+
+    public void func_406_a(Tessellator tessellator, float f, float f1, float f2, float f3, float f4, float f5) {
+        float f6 = (((float) field_667_e + f) / (float) field_666_f) * 32F;
+        if (f6 < 0.0F) {
+            f6 = 0.0F;
+        }
+        if (f6 > 1.0F) {
+            f6 = 1.0F;
+        }
+        field_665_g = field_671_a * f6;
+        super.func_406_a(tessellator, f, f1, f2, f3, f4, f5);
     }
 
     public void onUpdate() {
-        this.prevPosX = this.posX;
-        this.prevPosY = this.posY;
-        this.prevPosZ = this.posZ;
-        if(this.e++ >= this.field_666_f) {
-            this.setEntityDead();
-        }
-
-        this.field_670_b = 7 - this.e * 8 / this.field_666_f;
-        this.motionY += 0.004D;
-        this.moveEntity(this.motionX, this.motionY, this.motionZ);
-        if(this.posY == this.prevPosY) {
-            this.motionX *= 1.1D;
-            this.motionZ *= 1.1D;
-        }
-
-        this.motionX *= (double)0.96F;
-        this.motionY *= (double)0.96F;
-        this.motionZ *= (double)0.96F;
-        if(this.onGround) {
-            this.motionX *= (double)0.7F;
-            this.motionZ *= (double)0.7F;
-        }
-
+        prevPosX = posX;
+        prevPosY = posY;
+        prevPosZ = posZ;
+        if (field_667_e++ >= field_666_f) {
+            setEntityDead();
+        }
+        field_670_b = 7 - (field_667_e * 8) / field_666_f;
+        motionY += 0.0040000000000000001D;
+        moveEntity(motionX, motionY, motionZ);
+        if (posY == prevPosY) {
+            motionX *= 1.1000000000000001D;
+            motionZ *= 1.1000000000000001D;
+        }
+        motionX *= 0.95999997854232788D;
+        motionY *= 0.95999997854232788D;
+        motionZ *= 0.95999997854232788D;
+        if (onGround) {
+            motionX *= 0.69999998807907104D;
+            motionZ *= 0.69999998807907104D;
+        }
     }
 }
