--- net/minecraft/src/EntityExplodeFX.java
+++ net/minecraft/src/EntityExplodeFX.java
@@ -1,38 +1,38 @@
 package net.minecraft.src;
 
 public class EntityExplodeFX extends EntityFX {
-    public EntityExplodeFX(World var1, double var2, double var4, double var6, double var8, double var10, double var12) {
-        super(var1, var2, var4, var6, var8, var10, var12);
-        this.motionX = var8 + (double)((float)(Math.random() * 2.0D - 1.0D) * 0.05F);
-        this.motionY = var10 + (double)((float)(Math.random() * 2.0D - 1.0D) * 0.05F);
-        this.motionZ = var12 + (double)((float)(Math.random() * 2.0D - 1.0D) * 0.05F);
-        this.particleRed = this.particleBlue = this.particleGreen = this.rand.nextFloat() * 0.3F + 0.7F;
-        this.field_665_g = this.rand.nextFloat() * this.rand.nextFloat() * 6.0F + 1.0F;
-        this.field_666_f = (int)(16.0D / ((double)this.rand.nextFloat() * 0.8D + 0.2D)) + 2;
+
+    public EntityExplodeFX(World world, double d, double d1, double d2,
+                           double d3, double d4, double d5) {
+        super(world, d, d1, d2, d3, d4, d5);
+        motionX = d3 + (double) ((float) (Math.random() * 2D - 1.0D) * 0.05F);
+        motionY = d4 + (double) ((float) (Math.random() * 2D - 1.0D) * 0.05F);
+        motionZ = d5 + (double) ((float) (Math.random() * 2D - 1.0D) * 0.05F);
+        particleRed = particleBlue = particleGreen = rand.nextFloat() * 0.3F + 0.7F;
+        field_665_g = rand.nextFloat() * rand.nextFloat() * 6F + 1.0F;
+        field_666_f = (int) (16D / ((double) rand.nextFloat() * 0.80000000000000004D + 0.20000000000000001D)) + 2;
     }
 
-    public void func_406_a(Tessellator var1, float var2, float var3, float var4, float var5, float var6, float var7) {
-        super.func_406_a(var1, var2, var3, var4, var5, var6, var7);
+    public void func_406_a(Tessellator tessellator, float f, float f1, float f2, float f3, float f4, float f5) {
+        super.func_406_a(tessellator, f, f1, f2, f3, f4, f5);
     }
 
     public void onUpdate() {
-        this.prevPosX = this.posX;
-        this.prevPosY = this.posY;
-        this.prevPosZ = this.posZ;
-        if(this.e++ >= this.field_666_f) {
-            this.setEntityDead();
-        }
-
-        this.field_670_b = 7 - this.e * 8 / this.field_666_f;
-        this.motionY += 0.004D;
-        this.moveEntity(this.motionX, this.motionY, this.motionZ);
-        this.motionX *= (double)0.9F;
-        this.motionY *= (double)0.9F;
-        this.motionZ *= (double)0.9F;
-        if(this.onGround) {
-            this.motionX *= (double)0.7F;
-            this.motionZ *= (double)0.7F;
-        }
-
+        prevPosX = posX;
+        prevPosY = posY;
+        prevPosZ = posZ;
+        if (field_667_e++ >= field_666_f) {
+            setEntityDead();
+        }
+        field_670_b = 7 - (field_667_e * 8) / field_666_f;
+        motionY += 0.0040000000000000001D;
+        moveEntity(motionX, motionY, motionZ);
+        motionX *= 0.89999997615814209D;
+        motionY *= 0.89999997615814209D;
+        motionZ *= 0.89999997615814209D;
+        if (onGround) {
+            motionX *= 0.69999998807907104D;
+            motionZ *= 0.69999998807907104D;
+        }
     }
 }
