--- net/minecraft/src/FontAllowedCharacters.java
+++ net/minecraft/src/FontAllowedCharacters.java
@@ -4,29 +4,30 @@
 import java.io.InputStreamReader;
 
 public class FontAllowedCharacters {
+
     public static final String field_20157_a = func_20156_a();
 
+    public FontAllowedCharacters() {
+    }
+
     private static String func_20156_a() {
-        String var0 = "";
-
+        String s = "";
         try {
-            BufferedReader var1 = new BufferedReader(new InputStreamReader(FontAllowedCharacters.class.getResourceAsStream("/font.txt"), "UTF-8"));
-            String var2 = "";
-
-            while(true) {
-                var2 = var1.readLine();
-                if(var2 == null) {
-                    var1.close();
+            BufferedReader bufferedreader = new BufferedReader(new InputStreamReader((FontAllowedCharacters.class).getResourceAsStream("/font.txt"), "UTF-8"));
+            String s1 = "";
+            do {
+                String s2;
+                if ((s2 = bufferedreader.readLine()) == null) {
                     break;
                 }
-
-                if(!var2.startsWith("#")) {
-                    var0 = var0 + var2;
+                if (!s2.startsWith("#")) {
+                    s = (new StringBuilder()).append(s).append(s2).toString();
                 }
-            }
-        } catch (Exception var3) {
+            } while (true);
+            bufferedreader.close();
+        } catch (Exception exception) {
         }
-
-        return var0;
+        return s;
     }
+
 }
