--- net/minecraft/src/FontRenderer.java
+++ net/minecraft/src/FontRenderer.java
@@ -6,6 +6,9 @@
 import javax.imageio.ImageIO;
 import org.lwjgl.opengl.GL11;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.font.DrawStringEvent;
+
 public class FontRenderer {
     private int[] charWidth = new int[256];
     public int fontTextureName = 0;
@@ -123,6 +126,13 @@
 
     public void renderString(String var1, int var2, int var3, int var4, boolean var5) {
         if(var1 != null) {
+            DrawStringEvent event = new DrawStringEvent(var1);
+            Ref.getEventManager().handleEvent(event);
+
+            if (event.isCancelled()) {
+                var1 = event.getString();
+            }
+
             int var6;
             if(var5) {
                 var6 = var4 & -16777216;
