--- net/minecraft/src/GuiChat.java
+++ net/minecraft/src/GuiChat.java
@@ -2,11 +2,25 @@
 
 import org.lwjgl.input.Keyboard;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.gui.chat.ChatBackgroundDrawEvent;
+import cc.noxiuam.titanic.event.impl.gui.chat.PreChatMessageUpdateEvent;
+
 public class GuiChat extends GuiScreen {
     private String message = "";
     private int updateCounter = 0;
     private static final String field_20082_i = FontAllowedCharacters.field_20157_a;
 
+    public GuiChat(boolean command) {
+        message = "";
+
+        if (command) {
+            message = "/";
+        }
+
+        updateCounter = 0;
+    }
+
     public void initGui() {
         Keyboard.enableRepeatEvents(true);
     }
@@ -39,10 +53,22 @@
             }
 
         }
+
+        PreChatMessageUpdateEvent event = new PreChatMessageUpdateEvent(var2, message);
+        Ref.getEventManager().handleEvent(event);
+
+        if (event.isCancelled()) {
+            message = event.getMessage();
+        }
     }
 
     public void drawScreen(int var1, int var2, float var3) {
-        this.drawRect(2, this.height - 14, this.width - 2, this.height - 2, Integer.MIN_VALUE);
+        ChatBackgroundDrawEvent chatBackgroundDrawEvent = new ChatBackgroundDrawEvent();
+        Ref.getEventManager().handleEvent(chatBackgroundDrawEvent);
+
+        if (!chatBackgroundDrawEvent.isCancelled()) {
+            drawRect(2, height - 14, width - 2, height - 2, 0x80000000);
+        }
         this.drawString(this.fontRenderer, "> " + this.message + (this.updateCounter / 6 % 2 == 0 ? "_" : ""), 4, this.height - 12, 14737632);
     }
 
