--- net/minecraft/src/GuiConflictWarning.java
+++ net/minecraft/src/GuiConflictWarning.java
@@ -1,33 +1,38 @@
 package net.minecraft.src;
 
 public class GuiConflictWarning extends GuiScreen {
-    private int updateCounter = 0;
+
+    private int updateCounter;
+
+    public GuiConflictWarning() {
+        updateCounter = 0;
+    }
 
     public void updateScreen() {
-        ++this.updateCounter;
+        updateCounter++;
     }
 
     public void initGui() {
-        this.controlList.clear();
-        this.controlList.add(new GuiButton(0, this.width / 2 - 100, this.height / 4 + 120 + 12, "Back to title screen"));
-    }
-
-    protected void actionPerformed(GuiButton var1) {
-        if(var1.enabled) {
-            if(var1.id == 0) {
-                this.mc.displayGuiScreen(new GuiMainMenu());
-            }
-
-        }
-    }
-
-    public void drawScreen(int var1, int var2, float var3) {
-        this.drawDefaultBackground();
-        this.drawCenteredString(this.fontRenderer, "Level save conflict", this.width / 2, this.height / 4 - 60 + 20, 16777215);
-        this.drawString(this.fontRenderer, "Minecraft detected a conflict in the level save data.", this.width / 2 - 140, this.height / 4 - 60 + 60 + 0, 10526880);
-        this.drawString(this.fontRenderer, "This could be caused by two copies of the game", this.width / 2 - 140, this.height / 4 - 60 + 60 + 18, 10526880);
-        this.drawString(this.fontRenderer, "accessing the same level.", this.width / 2 - 140, this.height / 4 - 60 + 60 + 27, 10526880);
-        this.drawString(this.fontRenderer, "To prevent level corruption, the current game has quit.", this.width / 2 - 140, this.height / 4 - 60 + 60 + 45, 10526880);
-        super.drawScreen(var1, var2, var3);
+        controlList.clear();
+        controlList.add(new GuiButton(0, width / 2 - 100, height / 4 + 120 + 12, "Back to title screen"));
+    }
+
+    protected void actionPerformed(GuiButton guibutton) {
+        if (!guibutton.enabled) {
+            return;
+        }
+        if (guibutton.id == 0) {
+            mc.displayGuiScreen(new GuiMainMenu());
+        }
+    }
+
+    public void drawScreen(int i, int j, float f) {
+        drawDefaultBackground();
+        drawCenteredString(fontRenderer, "Level save conflict", width / 2, (height / 4 - 60) + 20, 0xffffff);
+        drawString(fontRenderer, "Minecraft detected a conflict in the level save data.", width / 2 - 140, (height / 4 - 60) + 60, 0xa0a0a0);
+        drawString(fontRenderer, "This could be caused by two copies of the game", width / 2 - 140, (height / 4 - 60) + 60 + 18, 0xa0a0a0);
+        drawString(fontRenderer, "accessing the same level.", width / 2 - 140, (height / 4 - 60) + 60 + 27, 0xa0a0a0);
+        drawString(fontRenderer, "To prevent level corruption, the current game has quit.", width / 2 - 140, (height / 4 - 60) + 60 + 45, 0xa0a0a0);
+        super.drawScreen(i, j, f);
     }
 }
