--- net/minecraft/src/GuiContainer.java
+++ net/minecraft/src/GuiContainer.java
@@ -3,6 +3,9 @@
 import org.lwjgl.opengl.GL11;
 import org.lwjgl.opengl.GL12;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.gui.SlotChangeEvent;
+
 public abstract class GuiContainer extends GuiScreen {
     private static RenderItem itemRenderer = new RenderItem();
     protected int xSize = 176;
@@ -137,7 +140,15 @@
             }
 
             if(var8 != -1) {
-                this.mc.playerController.func_20085_a(this.inventorySlots.unusedList, var8, var3, this.mc.thePlayer);
+                // TODO: Redo this mod's implementation.
+                SlotChangeEvent event = new SlotChangeEvent();
+                Ref.getEventManager().handleEvent(event);
+
+                if (event.isCancelled()) {
+                    Ref.getModuleManager().getModernInventoryModule().handleClickOnSlot(var8, var3, this.mc, this.inventorySlots);
+                } else {
+                    mc.playerController.func_20085_a(inventorySlots.unusedList, var8, var3, mc.thePlayer);
+                }
             }
         }
 
