--- net/minecraft/src/GuiIngameMenu.java
+++ net/minecraft/src/GuiIngameMenu.java
@@ -1,58 +1,60 @@
 package net.minecraft.src;
 
 public class GuiIngameMenu extends GuiScreen {
-    private int updateCounter2 = 0;
-    private int updateCounter = 0;
+
+    private int updateCounter2;
+    private int updateCounter;
+
+    public GuiIngameMenu() {
+        updateCounter2 = 0;
+        updateCounter = 0;
+    }
 
     public void initGui() {
-        this.updateCounter2 = 0;
-        this.controlList.clear();
-        this.controlList.add(new GuiButton(1, this.width / 2 - 100, this.height / 4 + 48, "Save and quit to title"));
-        if(this.mc.isMultiplayerWorld()) {
-            ((GuiButton)this.controlList.get(0)).displayString = "Disconnect";
+        updateCounter2 = 0;
+        controlList.clear();
+        controlList.add(new GuiButton(1, width / 2 - 100, height / 4 + 48, "Save and quit to title"));
+        if (mc.isMultiplayerWorld()) {
+            ((GuiButton) controlList.get(0)).displayString = "Disconnect";
         }
-
-        this.controlList.add(new GuiButton(4, this.width / 2 - 100, this.height / 4 + 24, "Back to game"));
-        this.controlList.add(new GuiButton(0, this.width / 2 - 100, this.height / 4 + 96, "Options..."));
+        controlList.add(new GuiButton(4, width / 2 - 100, height / 4 + 24, "Back to game"));
+        controlList.add(new GuiButton(0, width / 2 - 100, height / 4 + 96, "Options..."));
     }
 
-    protected void actionPerformed(GuiButton var1) {
-        if(var1.id == 0) {
-            this.mc.displayGuiScreen(new GuiOptions(this, this.mc.gameSettings));
+    protected void actionPerformed(GuiButton guibutton) {
+        if (guibutton.id == 0) {
+            mc.displayGuiScreen(new GuiOptions(this, mc.gameSettings));
         }
-
-        if(var1.id == 1) {
-            if(this.mc.isMultiplayerWorld()) {
-                this.mc.theWorld.sendQuittingDisconnectingPacket();
+        if (guibutton.id == 1) {
+            if (mc.isMultiplayerWorld()) {
+                mc.theWorld.sendQuittingDisconnectingPacket();
             }
-
-            this.mc.func_6261_a((World)null);
-            this.mc.displayGuiScreen(new GuiMainMenu());
-        }
-
-        if(var1.id == 4) {
-            this.mc.displayGuiScreen((GuiScreen)null);
-            this.mc.func_6259_e();
-        }
-
+            mc.func_6261_a(null);
+            mc.displayGuiScreen(new GuiMainMenu());
+        }
+        if (guibutton.id == 4) {
+            mc.displayGuiScreen(null);
+            mc.func_6259_e();
+        }
     }
 
     public void updateScreen() {
         super.updateScreen();
-        ++this.updateCounter;
+        updateCounter++;
     }
 
-    public void drawScreen(int var1, int var2, float var3) {
-        this.drawDefaultBackground();
-        boolean var4 = !this.mc.theWorld.func_650_a(this.updateCounter2++);
-        if(var4 || this.updateCounter < 20) {
-            float var5 = ((float)(this.updateCounter % 10) + var3) / 10.0F;
-            var5 = MathHelper.sin(var5 * (float)Math.PI * 2.0F) * 0.2F + 0.8F;
-            int var6 = (int)(255.0F * var5);
-            this.drawString(this.fontRenderer, "Saving level..", 8, this.height - 16, var6 << 16 | var6 << 8 | var6);
+    public void drawScreen(int i, int j, float f) {
+        drawDefaultBackground();
+        boolean flag = !mc.theWorld.func_650_a(updateCounter2++);
+
+        if (flag || updateCounter < 20) {
+            float f1 = ((float) (updateCounter % 10) + f) / 10F;
+            f1 = MathHelper.sin(f1 * 3.141593F * 2.0F) * 0.2F + 0.8F;
+            int k = (int) (255F * f1);
+            drawString(fontRenderer, "Saving level..", 8, height - 16, k << 16 | k << 8 | k);
         }
 
-        this.drawCenteredString(this.fontRenderer, "Game menu", this.width / 2, 40, 16777215);
-        super.drawScreen(var1, var2, var3);
+        drawCenteredString(fontRenderer, "Game menu", width / 2, 40, 0xffffff);
+        super.drawScreen(i, j, f);
     }
 }
