--- net/minecraft/src/IBlockAccess.java
+++ net/minecraft/src/IBlockAccess.java
@@ -1,17 +1,18 @@
 package net.minecraft.src;
 
 public interface IBlockAccess {
-    int getBlockId(int var1, int var2, int var3);
-
-    TileEntity getBlockTileEntity(int var1, int var2, int var3);
-
-    float getLightBrightness(int var1, int var2, int var3);
-
-    int getBlockMetadata(int var1, int var2, int var3);
-
-    Material getBlockMaterial(int var1, int var2, int var3);
-
-    boolean isBlockOpaqueCube(int var1, int var2, int var3);
-
-    WorldChunkManager func_4075_a();
+
+    public abstract int getBlockId(int i, int j, int k);
+
+    public abstract TileEntity getBlockTileEntity(int i, int j, int k);
+
+    public abstract float getLightBrightness(int i, int j, int k);
+
+    public abstract int getBlockMetadata(int i, int j, int k);
+
+    public abstract Material getBlockMaterial(int i, int j, int k);
+
+    public abstract boolean isBlockOpaqueCube(int i, int j, int k);
+
+    public abstract WorldChunkManager func_4075_a();
 }
