--- net/minecraft/src/IChunkLoader.java
+++ net/minecraft/src/IChunkLoader.java
@@ -3,13 +3,14 @@
 import java.io.IOException;
 
 public interface IChunkLoader {
-    Chunk loadChunk(World var1, int var2, int var3) throws IOException;
-
-    void saveChunk(World var1, Chunk var2) throws IOException;
-
-    void saveExtraChunkData(World var1, Chunk var2) throws IOException;
-
-    void func_814_a();
-
-    void saveExtraData();
+
+    public abstract Chunk loadChunk(World world, int i, int j) throws IOException;
+
+    public abstract void saveChunk(World world, Chunk chunk) throws IOException;
+
+    public abstract void saveExtraChunkData(World world, Chunk chunk) throws IOException;
+
+    public abstract void func_814_a();
+
+    public abstract void saveExtraData();
 }
