--- net/minecraft/src/InventoryCraftResult.java
+++ net/minecraft/src/InventoryCraftResult.java
@@ -1,32 +1,38 @@
 package net.minecraft.src;
 
-public class InventoryCraftResult implements IInventory {
-    private ItemStack[] stackResult = new ItemStack[1];
+public class InventoryCraftResult
+        implements IInventory {
+
+    private ItemStack stackResult[];
+
+    public InventoryCraftResult() {
+        stackResult = new ItemStack[1];
+    }
 
     public int getSizeInventory() {
         return 1;
     }
 
-    public ItemStack getStackInSlot(int var1) {
-        return this.stackResult[var1];
+    public ItemStack getStackInSlot(int i) {
+        return stackResult[i];
     }
 
     public String getInvName() {
         return "Result";
     }
 
-    public ItemStack decrStackSize(int var1, int var2) {
-        if(this.stackResult[var1] != null) {
-            ItemStack var3 = this.stackResult[var1];
-            this.stackResult[var1] = null;
-            return var3;
+    public ItemStack decrStackSize(int i, int j) {
+        if (stackResult[i] != null) {
+            ItemStack itemstack = stackResult[i];
+            stackResult[i] = null;
+            return itemstack;
         } else {
             return null;
         }
     }
 
-    public void setInventorySlotContents(int var1, ItemStack var2) {
-        this.stackResult[var1] = var2;
+    public void setInventorySlotContents(int i, ItemStack itemstack) {
+        stackResult[i] = itemstack;
     }
 
     public int getInventoryStackLimit() {
@@ -36,7 +42,7 @@
     public void onInventoryChanged() {
     }
 
-    public boolean func_20070_a_(EntityPlayer var1) {
+    public boolean func_20070_a_(EntityPlayer entityplayer) {
         return true;
     }
 }
