--- net/minecraft/src/IProgressUpdate.java
+++ net/minecraft/src/IProgressUpdate.java
@@ -1,9 +1,10 @@
 package net.minecraft.src;
 
 public interface IProgressUpdate {
-    void func_594_b(String var1);
-
-    void displayLoadingString(String var1);
-
-    void setLoadingProgress(int var1);
+
+    void func_594_b(String s);
+
+    void displayLoadingString(String s);
+
+    void setLoadingProgress(int i);
 }
