--- net/minecraft/src/ItemEgg.java
+++ net/minecraft/src/ItemEgg.java
@@ -1,18 +1,18 @@
 package net.minecraft.src;
 
 public class ItemEgg extends Item {
-    public ItemEgg(int var1) {
-        super(var1);
-        this.maxStackSize = 16;
+
+    public ItemEgg(int i) {
+        super(i);
+        maxStackSize = 16;
     }
 
-    public ItemStack onItemRightClick(ItemStack var1, World var2, EntityPlayer var3) {
-        --var1.stackSize;
-        var2.playSoundAtEntity(var3, "random.bow", 0.5F, 0.4F / (itemRand.nextFloat() * 0.4F + 0.8F));
-        if(!var2.multiplayerWorld) {
-            var2.entityJoinedWorld(new EntityEgg(var2, var3));
+    public ItemStack onItemRightClick(ItemStack itemstack, World world, EntityPlayer entityplayer) {
+        itemstack.stackSize--;
+        world.playSoundAtEntity(entityplayer, "random.bow", 0.5F, 0.4F / (itemRand.nextFloat() * 0.4F + 0.8F));
+        if (!world.multiplayerWorld) {
+            world.entityJoinedWorld(new EntityEgg(world, entityplayer));
         }
-
-        return var1;
+        return itemstack;
     }
 }
