--- net/minecraft/src/ItemSaddle.java
+++ net/minecraft/src/ItemSaddle.java
@@ -1,24 +1,24 @@
 package net.minecraft.src;
 
 public class ItemSaddle extends Item {
-    public ItemSaddle(int var1) {
-        super(var1);
-        this.maxStackSize = 1;
-        this.maxDamage = 64;
+
+    public ItemSaddle(int i) {
+        super(i);
+        maxStackSize = 1;
+        maxDamage = 64;
     }
 
-    public void func_4019_b(ItemStack var1, EntityLiving var2) {
-        if(var2 instanceof EntityPig) {
-            EntityPig var3 = (EntityPig)var2;
-            if(!var3.rideable) {
-                var3.rideable = true;
-                --var1.stackSize;
+    public void func_4019_b(ItemStack itemstack, EntityLiving entityliving) {
+        if (entityliving instanceof EntityPig) {
+            EntityPig entitypig = (EntityPig) entityliving;
+            if (!entitypig.rideable) {
+                entitypig.rideable = true;
+                itemstack.stackSize--;
             }
         }
-
     }
 
-    public void hitEntity(ItemStack var1, EntityLiving var2) {
-        this.func_4019_b(var1, var2);
+    public void hitEntity(ItemStack itemstack, EntityLiving entityliving) {
+        func_4019_b(itemstack, entityliving);
     }
 }
