--- net/minecraft/src/ItemStack.java
+++ net/minecraft/src/ItemStack.java
@@ -134,7 +134,7 @@
         return var0 == null && var1 == null ? true : (var0 != null && var1 != null ? var0.func_20108_a(var1) : false);
     }
 
-    private boolean func_20108_a(ItemStack var1) {
+    public boolean func_20108_a(ItemStack var1) {
         return this.stackSize != var1.stackSize ? false : (this.itemID != var1.itemID ? false : this.itemDamage == var1.itemDamage);
     }
 
