--- net/minecraft/src/ItemAxe.java
+++ net/minecraft/src/ItemAxe.java
@@ -1,9 +1,16 @@
 package net.minecraft.src;
 
 public class ItemAxe extends ItemTool {
-    private static Block[] blocksEffectiveAgainst = new Block[]{Block.planks, Block.bookShelf, Block.wood, Block.crate};
-
-    public ItemAxe(int var1, int var2) {
-        super(var1, 3, var2, blocksEffectiveAgainst);
+
+    private static Block blocksEffectiveAgainst[];
+
+    static {
+        blocksEffectiveAgainst = (new Block[]{
+                Block.planks, Block.bookShelf, Block.wood, Block.crate
+        });
+    }
+
+    public ItemAxe(int i, int j) {
+        super(i, 3, j, blocksEffectiveAgainst);
     }
 }
