--- net/minecraft/src/ItemSeeds.java
+++ net/minecraft/src/ItemSeeds.java
@@ -1,25 +1,25 @@
 package net.minecraft.src;
 
 public class ItemSeeds extends Item {
+
     private int field_318_a;
 
-    public ItemSeeds(int var1, int var2) {
-        super(var1);
-        this.field_318_a = var2;
+    public ItemSeeds(int i, int j) {
+        super(i);
+        field_318_a = j;
     }
 
-    public boolean onItemUse(ItemStack var1, EntityPlayer var2, World var3, int var4, int var5, int var6, int var7) {
-        if(var7 != 1) {
+    public boolean onItemUse(ItemStack itemstack, EntityPlayer entityplayer, World world, int i, int j, int k, int l) {
+        if (l != 1) {
             return false;
+        }
+        int i1 = world.getBlockId(i, j, k);
+        if (i1 == Block.tilledField.blockID) {
+            world.setBlockWithNotify(i, j + 1, k, field_318_a);
+            itemstack.stackSize--;
+            return true;
         } else {
-            int var8 = var3.getBlockId(var4, var5, var6);
-            if(var8 == Block.tilledField.blockID) {
-                var3.setBlockWithNotify(var4, var5 + 1, var6, this.field_318_a);
-                --var1.stackSize;
-                return true;
-            } else {
-                return false;
-            }
+            return false;
         }
     }
 }
