--- net/minecraft/src/ItemBlock.java
+++ net/minecraft/src/ItemBlock.java
@@ -1,61 +1,54 @@
 package net.minecraft.src;
 
 public class ItemBlock extends Item {
+
     private int blockID;
 
-    public ItemBlock(int var1) {
-        super(var1);
-        this.blockID = var1 + 256;
-        this.setIconIndex(Block.blocksList[var1 + 256].getBlockTextureFromSide(2));
+    public ItemBlock(int i) {
+        super(i);
+        blockID = i + 256;
+        setIconIndex(Block.blocksList[i + 256].getBlockTextureFromSide(2));
     }
 
-    public boolean onItemUse(ItemStack var1, EntityPlayer var2, World var3, int var4, int var5, int var6, int var7) {
-        if(var3.getBlockId(var4, var5, var6) == Block.snow.blockID) {
-            var7 = 0;
+    public boolean onItemUse(ItemStack itemstack, EntityPlayer entityplayer, World world, int i, int j, int k, int l) {
+        if (world.getBlockId(i, j, k) == Block.snow.blockID) {
+            l = 0;
         } else {
-            if(var7 == 0) {
-                --var5;
-            }
-
-            if(var7 == 1) {
-                ++var5;
-            }
-
-            if(var7 == 2) {
-                --var6;
-            }
-
-            if(var7 == 3) {
-                ++var6;
-            }
-
-            if(var7 == 4) {
-                --var4;
-            }
-
-            if(var7 == 5) {
-                ++var4;
+            if (l == 0) {
+                j--;
+            }
+            if (l == 1) {
+                j++;
+            }
+            if (l == 2) {
+                k--;
+            }
+            if (l == 3) {
+                k++;
+            }
+            if (l == 4) {
+                i--;
+            }
+            if (l == 5) {
+                i++;
             }
         }
-
-        if(var1.stackSize == 0) {
+        if (itemstack.stackSize == 0) {
             return false;
-        } else {
-            if(var3.canBlockBePlacedAt(this.blockID, var4, var5, var6, false)) {
-                Block var8 = Block.blocksList[this.blockID];
-                if(var3.setBlockWithNotify(var4, var5, var6, this.blockID)) {
-                    Block.blocksList[this.blockID].onBlockPlaced(var3, var4, var5, var6, var7);
-                    Block.blocksList[this.blockID].onBlockPlacedBy(var3, var4, var5, var6, var2);
-                    var3.playSoundEffect((double)((float)var4 + 0.5F), (double)((float)var5 + 0.5F), (double)((float)var6 + 0.5F), var8.stepSound.func_1145_d(), (var8.stepSound.func_1147_b() + 1.0F) / 2.0F, var8.stepSound.func_1144_c() * 0.8F);
-                    --var1.stackSize;
-                }
+        }
+        if (world.canBlockBePlacedAt(blockID, i, j, k, false)) {
+            Block block = Block.blocksList[blockID];
+            if (world.setBlockWithNotify(i, j, k, blockID)) {
+                Block.blocksList[blockID].onBlockPlaced(world, i, j, k, l);
+                Block.blocksList[blockID].onBlockPlacedBy(world, i, j, k, entityplayer);
+                world.playSoundEffect((float) i + 0.5F, (float) j + 0.5F, (float) k + 0.5F, block.stepSound.func_1145_d(), (block.stepSound.func_1147_b() + 1.0F) / 2.0F, block.stepSound.func_1144_c() * 0.8F);
+                itemstack.stackSize--;
             }
-
-            return true;
         }
+        return true;
     }
 
     public String func_20009_a() {
-        return Block.blocksList[this.blockID].func_20013_i();
+        return Block.blocksList[blockID].func_20013_i();
     }
 }
