--- net/minecraft/src/ItemRenderer.java
+++ net/minecraft/src/ItemRenderer.java
@@ -4,6 +4,9 @@
 import org.lwjgl.opengl.GL11;
 import org.lwjgl.opengl.GL12;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.gui.FirstPersonFireDrawEvent;
+
 public class ItemRenderer {
     private Minecraft mc;
     private ItemStack field_9451_b = null;
@@ -284,6 +287,13 @@
     }
 
     private void renderFireInFirstPerson(float var1) {
+        FirstPersonFireDrawEvent event = new FirstPersonFireDrawEvent();
+        Ref.getEventManager().handleEvent(event);
+
+        if (event.isCancelled()) {
+            return;
+        }
+
         Tessellator var2 = Tessellator.instance;
         GL11.glColor4f(1.0F, 1.0F, 1.0F, 0.9F);
         GL11.glEnable(GL11.GL_BLEND);
