--- net/minecraft/src/KeyBinding.java
+++ net/minecraft/src/KeyBinding.java
@@ -1,11 +1,11 @@
 package net.minecraft.src;
 
 public class KeyBinding {
+
     public String keyDescription;
     public int keyCode;
-
-    public KeyBinding(String var1, int var2) {
-        this.keyDescription = var1;
-        this.keyCode = var2;
+    public KeyBinding(String s, int i) {
+        keyDescription = s;
+        keyCode = i;
     }
 }
