--- net/minecraft/src/Material.java
+++ net/minecraft/src/Material.java
@@ -1,6 +1,7 @@
 package net.minecraft.src;
 
 public class Material {
+
     public static final Material air = new MaterialTransparent();
     public static final Material ground = new Material();
     public static final Material wood = (new Material()).setBurning();
@@ -26,6 +27,8 @@
     public static final Material pumpkin = new Material();
     public static final Material portal = new Material();
     private boolean canBurn;
+    public Material() {
+    }
 
     public boolean getIsLiquid() {
         return false;
@@ -44,11 +47,12 @@
     }
 
     private Material setBurning() {
-        this.canBurn = true;
+        canBurn = true;
         return this;
     }
 
     public boolean getBurning() {
-        return this.canBurn;
+        return canBurn;
     }
+
 }
