--- net/minecraft/src/MetadataChunkBlock.java
+++ net/minecraft/src/MetadataChunkBlock.java
@@ -1,6 +1,7 @@
 package net.minecraft.src;
 
 public class MetadataChunkBlock {
+
     public final EnumSkyBlock field_1299_a;
     public int field_1298_b;
     public int field_1304_c;
@@ -8,168 +9,151 @@
     public int field_1302_e;
     public int field_1301_f;
     public int field_1300_g;
-
-    public MetadataChunkBlock(EnumSkyBlock var1, int var2, int var3, int var4, int var5, int var6, int var7) {
-        this.field_1299_a = var1;
-        this.field_1298_b = var2;
-        this.field_1304_c = var3;
-        this.field_1303_d = var4;
-        this.field_1302_e = var5;
-        this.field_1301_f = var6;
-        this.field_1300_g = var7;
+    public MetadataChunkBlock(EnumSkyBlock enumskyblock, int i, int j, int k, int l, int i1, int j1) {
+        field_1299_a = enumskyblock;
+        field_1298_b = i;
+        field_1304_c = j;
+        field_1303_d = k;
+        field_1302_e = l;
+        field_1301_f = i1;
+        field_1300_g = j1;
     }
 
-    public void func_4127_a(World var1) {
-        int var2 = this.field_1302_e - this.field_1298_b + 1;
-        int var3 = this.field_1301_f - this.field_1304_c + 1;
-        int var4 = this.field_1300_g - this.field_1303_d + 1;
-        int var5 = var2 * var3 * var4;
-        if(var5 <= -Short.MIN_VALUE) {
-            for(int var6 = this.field_1298_b; var6 <= this.field_1302_e; ++var6) {
-                for(int var7 = this.field_1303_d; var7 <= this.field_1300_g; ++var7) {
-                    if(var1.blockExists(var6, 0, var7)) {
-                        for(int var8 = this.field_1304_c; var8 <= this.field_1301_f; ++var8) {
-                            if(var8 >= 0 && var8 < 128) {
-                                int var9 = var1.getSavedLightValue(this.field_1299_a, var6, var8, var7);
-                                boolean var10 = false;
-                                int var11 = var1.getBlockId(var6, var8, var7);
-                                int var12 = Block.lightOpacity[var11];
-                                if(var12 == 0) {
-                                    var12 = 1;
-                                }
-
-                                int var13 = 0;
-                                if(this.field_1299_a == EnumSkyBlock.Sky) {
-                                    if(var1.canExistingBlockSeeTheSky(var6, var8, var7)) {
-                                        var13 = 15;
-                                    }
-                                } else if(this.field_1299_a == EnumSkyBlock.Block) {
-                                    var13 = Block.lightValue[var11];
-                                }
-
-                                int var14;
-                                int var20;
-                                if(var12 >= 15 && var13 == 0) {
-                                    var20 = 0;
-                                } else {
-                                    var14 = var1.getSavedLightValue(this.field_1299_a, var6 - 1, var8, var7);
-                                    int var15 = var1.getSavedLightValue(this.field_1299_a, var6 + 1, var8, var7);
-                                    int var16 = var1.getSavedLightValue(this.field_1299_a, var6, var8 - 1, var7);
-                                    int var17 = var1.getSavedLightValue(this.field_1299_a, var6, var8 + 1, var7);
-                                    int var18 = var1.getSavedLightValue(this.field_1299_a, var6, var8, var7 - 1);
-                                    int var19 = var1.getSavedLightValue(this.field_1299_a, var6, var8, var7 + 1);
-                                    var20 = var14;
-                                    if(var15 > var14) {
-                                        var20 = var15;
-                                    }
-
-                                    if(var16 > var20) {
-                                        var20 = var16;
-                                    }
-
-                                    if(var17 > var20) {
-                                        var20 = var17;
-                                    }
-
-                                    if(var18 > var20) {
-                                        var20 = var18;
-                                    }
-
-                                    if(var19 > var20) {
-                                        var20 = var19;
-                                    }
-
-                                    var20 -= var12;
-                                    if(var20 < 0) {
-                                        var20 = 0;
-                                    }
-
-                                    if(var13 > var20) {
-                                        var20 = var13;
-                                    }
-                                }
-
-                                if(var9 != var20) {
-                                    var1.setLightValue(this.field_1299_a, var6, var8, var7, var20);
-                                    var14 = var20 - 1;
-                                    if(var14 < 0) {
-                                        var14 = 0;
-                                    }
-
-                                    var1.neighborLightPropagationChanged(this.field_1299_a, var6 - 1, var8, var7, var14);
-                                    var1.neighborLightPropagationChanged(this.field_1299_a, var6, var8 - 1, var7, var14);
-                                    var1.neighborLightPropagationChanged(this.field_1299_a, var6, var8, var7 - 1, var14);
-                                    if(var6 + 1 >= this.field_1302_e) {
-                                        var1.neighborLightPropagationChanged(this.field_1299_a, var6 + 1, var8, var7, var14);
-                                    }
-
-                                    if(var8 + 1 >= this.field_1301_f) {
-                                        var1.neighborLightPropagationChanged(this.field_1299_a, var6, var8 + 1, var7, var14);
-                                    }
-
-                                    if(var7 + 1 >= this.field_1300_g) {
-                                        var1.neighborLightPropagationChanged(this.field_1299_a, var6, var8, var7 + 1, var14);
-                                    }
-                                }
-                            }
-                        }
-                    }
-                }
+    public void func_4127_a(World world) {
+        int i = (field_1302_e - field_1298_b) + 1;
+        int j = (field_1301_f - field_1304_c) + 1;
+        int k = (field_1300_g - field_1303_d) + 1;
+        int l = i * j * k;
+        if (l > 32768) {
+            return;
+        }
+        for (int i1 = field_1298_b; i1 <= field_1302_e; i1++) {
+            for (int j1 = field_1303_d; j1 <= field_1300_g; j1++) {
+                if (!world.blockExists(i1, 0, j1)) {
+                    continue;
+                }
+                for (int k1 = field_1304_c; k1 <= field_1301_f; k1++) {
+                    if (k1 < 0 || k1 >= 128) {
+                        continue;
+                    }
+                    int l1 = world.getSavedLightValue(field_1299_a, i1, k1, j1);
+                    int i2 = 0;
+                    int j2 = world.getBlockId(i1, k1, j1);
+                    int k2 = Block.lightOpacity[j2];
+                    if (k2 == 0) {
+                        k2 = 1;
+                    }
+                    int l2 = 0;
+                    if (field_1299_a == EnumSkyBlock.Sky) {
+                        if (world.canExistingBlockSeeTheSky(i1, k1, j1)) {
+                            l2 = 15;
+                        }
+                    } else if (field_1299_a == EnumSkyBlock.Block) {
+                        l2 = Block.lightValue[j2];
+                    }
+                    if (k2 >= 15 && l2 == 0) {
+                        i2 = 0;
+                    } else {
+                        int i3 = world.getSavedLightValue(field_1299_a, i1 - 1, k1, j1);
+                        int k3 = world.getSavedLightValue(field_1299_a, i1 + 1, k1, j1);
+                        int l3 = world.getSavedLightValue(field_1299_a, i1, k1 - 1, j1);
+                        int i4 = world.getSavedLightValue(field_1299_a, i1, k1 + 1, j1);
+                        int j4 = world.getSavedLightValue(field_1299_a, i1, k1, j1 - 1);
+                        int k4 = world.getSavedLightValue(field_1299_a, i1, k1, j1 + 1);
+                        i2 = i3;
+                        if (k3 > i2) {
+                            i2 = k3;
+                        }
+                        if (l3 > i2) {
+                            i2 = l3;
+                        }
+                        if (i4 > i2) {
+                            i2 = i4;
+                        }
+                        if (j4 > i2) {
+                            i2 = j4;
+                        }
+                        if (k4 > i2) {
+                            i2 = k4;
+                        }
+                        i2 -= k2;
+                        if (i2 < 0) {
+                            i2 = 0;
+                        }
+                        if (l2 > i2) {
+                            i2 = l2;
+                        }
+                    }
+                    if (l1 == i2) {
+                        continue;
+                    }
+                    world.setLightValue(field_1299_a, i1, k1, j1, i2);
+                    int j3 = i2 - 1;
+                    if (j3 < 0) {
+                        j3 = 0;
+                    }
+                    world.neighborLightPropagationChanged(field_1299_a, i1 - 1, k1, j1, j3);
+                    world.neighborLightPropagationChanged(field_1299_a, i1, k1 - 1, j1, j3);
+                    world.neighborLightPropagationChanged(field_1299_a, i1, k1, j1 - 1, j3);
+                    if (i1 + 1 >= field_1302_e) {
+                        world.neighborLightPropagationChanged(field_1299_a, i1 + 1, k1, j1, j3);
+                    }
+                    if (k1 + 1 >= field_1301_f) {
+                        world.neighborLightPropagationChanged(field_1299_a, i1, k1 + 1, j1, j3);
+                    }
+                    if (j1 + 1 >= field_1300_g) {
+                        world.neighborLightPropagationChanged(field_1299_a, i1, k1, j1 + 1, j3);
+                    }
+                }
+
             }
 
         }
+
     }
 
-    public boolean func_866_a(int var1, int var2, int var3, int var4, int var5, int var6) {
-        if(var1 >= this.field_1298_b && var2 >= this.field_1304_c && var3 >= this.field_1303_d && var4 <= this.field_1302_e && var5 <= this.field_1301_f && var6 <= this.field_1300_g) {
+    public boolean func_866_a(int i, int j, int k, int l, int i1, int j1) {
+        if (i >= field_1298_b && j >= field_1304_c && k >= field_1303_d && l <= field_1302_e && i1 <= field_1301_f && j1 <= field_1300_g) {
             return true;
-        } else {
-            byte var7 = 1;
-            if(var1 >= this.field_1298_b - var7 && var2 >= this.field_1304_c - var7 && var3 >= this.field_1303_d - var7 && var4 <= this.field_1302_e + var7 && var5 <= this.field_1301_f + var7 && var6 <= this.field_1300_g + var7) {
-                int var8 = this.field_1302_e - this.field_1298_b;
-                int var9 = this.field_1301_f - this.field_1304_c;
-                int var10 = this.field_1300_g - this.field_1303_d;
-                if(var1 > this.field_1298_b) {
-                    var1 = this.field_1298_b;
-                }
-
-                if(var2 > this.field_1304_c) {
-                    var2 = this.field_1304_c;
-                }
-
-                if(var3 > this.field_1303_d) {
-                    var3 = this.field_1303_d;
-                }
-
-                if(var4 < this.field_1302_e) {
-                    var4 = this.field_1302_e;
-                }
-
-                if(var5 < this.field_1301_f) {
-                    var5 = this.field_1301_f;
-                }
-
-                if(var6 < this.field_1300_g) {
-                    var6 = this.field_1300_g;
-                }
-
-                int var11 = var4 - var1;
-                int var12 = var5 - var2;
-                int var13 = var6 - var3;
-                int var14 = var8 * var9 * var10;
-                int var15 = var11 * var12 * var13;
-                if(var15 - var14 <= 2) {
-                    this.field_1298_b = var1;
-                    this.field_1304_c = var2;
-                    this.field_1303_d = var3;
-                    this.field_1302_e = var4;
-                    this.field_1301_f = var5;
-                    this.field_1300_g = var6;
-                    return true;
-                }
-            }
-
-            return false;
-        }
+        }
+        int k1 = 1;
+        if (i >= field_1298_b - k1 && j >= field_1304_c - k1 && k >= field_1303_d - k1 && l <= field_1302_e + k1 && i1 <= field_1301_f + k1 && j1 <= field_1300_g + k1) {
+            int l1 = field_1302_e - field_1298_b;
+            int i2 = field_1301_f - field_1304_c;
+            int j2 = field_1300_g - field_1303_d;
+            if (i > field_1298_b) {
+                i = field_1298_b;
+            }
+            if (j > field_1304_c) {
+                j = field_1304_c;
+            }
+            if (k > field_1303_d) {
+                k = field_1303_d;
+            }
+            if (l < field_1302_e) {
+                l = field_1302_e;
+            }
+            if (i1 < field_1301_f) {
+                i1 = field_1301_f;
+            }
+            if (j1 < field_1300_g) {
+                j1 = field_1300_g;
+            }
+            int k2 = l - i;
+            int l2 = i1 - j;
+            int i3 = j1 - k;
+            int j3 = l1 * i2 * j2;
+            int k3 = k2 * l2 * i3;
+            if (k3 - j3 <= 2) {
+                field_1298_b = i;
+                field_1304_c = j;
+                field_1303_d = k;
+                field_1302_e = l;
+                field_1301_f = i1;
+                field_1300_g = j1;
+                return true;
+            }
+        }
+        return false;
     }
 }
