--- net/minecraft/src/ModelCreeper.java
+++ net/minecraft/src/ModelCreeper.java
@@ -1,6 +1,7 @@
 package net.minecraft.src;
 
 public class ModelCreeper extends ModelBase {
+
     public ModelRenderer head;
     public ModelRenderer unusedModel;
     public ModelRenderer body;
@@ -8,49 +9,48 @@
     public ModelRenderer leg2;
     public ModelRenderer leg3;
     public ModelRenderer leg4;
-
     public ModelCreeper() {
-        float var1 = 0.0F;
-        byte var2 = 4;
-        this.head = new ModelRenderer(0, 0);
-        this.head.addBox(-4.0F, -8.0F, -4.0F, 8, 8, 8, var1);
-        this.head.setPosition(0.0F, (float)var2, 0.0F);
-        this.unusedModel = new ModelRenderer(32, 0);
-        this.unusedModel.addBox(-4.0F, -8.0F, -4.0F, 8, 8, 8, var1 + 0.5F);
-        this.unusedModel.setPosition(0.0F, (float)var2, 0.0F);
-        this.body = new ModelRenderer(16, 16);
-        this.body.addBox(-4.0F, 0.0F, -2.0F, 8, 12, 4, var1);
-        this.body.setPosition(0.0F, (float)var2, 0.0F);
-        this.leg1 = new ModelRenderer(0, 16);
-        this.leg1.addBox(-2.0F, 0.0F, -2.0F, 4, 6, 4, var1);
-        this.leg1.setPosition(-2.0F, (float)(12 + var2), 4.0F);
-        this.leg2 = new ModelRenderer(0, 16);
-        this.leg2.addBox(-2.0F, 0.0F, -2.0F, 4, 6, 4, var1);
-        this.leg2.setPosition(2.0F, (float)(12 + var2), 4.0F);
-        this.leg3 = new ModelRenderer(0, 16);
-        this.leg3.addBox(-2.0F, 0.0F, -2.0F, 4, 6, 4, var1);
-        this.leg3.setPosition(-2.0F, (float)(12 + var2), -4.0F);
-        this.leg4 = new ModelRenderer(0, 16);
-        this.leg4.addBox(-2.0F, 0.0F, -2.0F, 4, 6, 4, var1);
-        this.leg4.setPosition(2.0F, (float)(12 + var2), -4.0F);
-    }
-
-    public void render(float var1, float var2, float var3, float var4, float var5, float var6) {
-        this.setRotationAngles(var1, var2, var3, var4, var5, var6);
-        this.head.render(var6);
-        this.body.render(var6);
-        this.leg1.render(var6);
-        this.leg2.render(var6);
-        this.leg3.render(var6);
-        this.leg4.render(var6);
-    }
-
-    public void setRotationAngles(float var1, float var2, float var3, float var4, float var5, float var6) {
-        this.head.rotateAngleY = var4 / (180.0F / (float)Math.PI);
-        this.head.rotateAngleX = var5 / (180.0F / (float)Math.PI);
-        this.leg1.rotateAngleX = MathHelper.cos(var1 * 0.6662F) * 1.4F * var2;
-        this.leg2.rotateAngleX = MathHelper.cos(var1 * 0.6662F + (float)Math.PI) * 1.4F * var2;
-        this.leg3.rotateAngleX = MathHelper.cos(var1 * 0.6662F + (float)Math.PI) * 1.4F * var2;
-        this.leg4.rotateAngleX = MathHelper.cos(var1 * 0.6662F) * 1.4F * var2;
+        float f = 0.0F;
+        int i = 4;
+        head = new ModelRenderer(0, 0);
+        head.addBox(-4F, -8F, -4F, 8, 8, 8, f);
+        head.setPosition(0.0F, i, 0.0F);
+        unusedModel = new ModelRenderer(32, 0);
+        unusedModel.addBox(-4F, -8F, -4F, 8, 8, 8, f + 0.5F);
+        unusedModel.setPosition(0.0F, i, 0.0F);
+        body = new ModelRenderer(16, 16);
+        body.addBox(-4F, 0.0F, -2F, 8, 12, 4, f);
+        body.setPosition(0.0F, i, 0.0F);
+        leg1 = new ModelRenderer(0, 16);
+        leg1.addBox(-2F, 0.0F, -2F, 4, 6, 4, f);
+        leg1.setPosition(-2F, 12 + i, 4F);
+        leg2 = new ModelRenderer(0, 16);
+        leg2.addBox(-2F, 0.0F, -2F, 4, 6, 4, f);
+        leg2.setPosition(2.0F, 12 + i, 4F);
+        leg3 = new ModelRenderer(0, 16);
+        leg3.addBox(-2F, 0.0F, -2F, 4, 6, 4, f);
+        leg3.setPosition(-2F, 12 + i, -4F);
+        leg4 = new ModelRenderer(0, 16);
+        leg4.addBox(-2F, 0.0F, -2F, 4, 6, 4, f);
+        leg4.setPosition(2.0F, 12 + i, -4F);
+    }
+
+    public void render(float f, float f1, float f2, float f3, float f4, float f5) {
+        setRotationAngles(f, f1, f2, f3, f4, f5);
+        head.render(f5);
+        body.render(f5);
+        leg1.render(f5);
+        leg2.render(f5);
+        leg3.render(f5);
+        leg4.render(f5);
+    }
+
+    public void setRotationAngles(float f, float f1, float f2, float f3, float f4, float f5) {
+        head.rotateAngleY = f3 / 57.29578F;
+        head.rotateAngleX = f4 / 57.29578F;
+        leg1.rotateAngleX = MathHelper.cos(f * 0.6662F) * 1.4F * f1;
+        leg2.rotateAngleX = MathHelper.cos(f * 0.6662F + 3.141593F) * 1.4F * f1;
+        leg3.rotateAngleX = MathHelper.cos(f * 0.6662F + 3.141593F) * 1.4F * f1;
+        leg4.rotateAngleX = MathHelper.cos(f * 0.6662F) * 1.4F * f1;
     }
 }
