--- net/minecraft/src/ModelQuadraped.java
+++ net/minecraft/src/ModelQuadraped.java
@@ -1,50 +1,51 @@
 package net.minecraft.src;
 
 public class ModelQuadraped extends ModelBase {
-    public ModelRenderer head = new ModelRenderer(0, 0);
+
+    public ModelRenderer head;
     public ModelRenderer body;
     public ModelRenderer leg1;
     public ModelRenderer leg2;
     public ModelRenderer leg3;
     public ModelRenderer leg4;
-
-    public ModelQuadraped(int var1, float var2) {
-        this.head.addBox(-4.0F, -4.0F, -8.0F, 8, 8, 8, var2);
-        this.head.setPosition(0.0F, (float)(18 - var1), -6.0F);
-        this.body = new ModelRenderer(28, 8);
-        this.body.addBox(-5.0F, -10.0F, -7.0F, 10, 16, 8, var2);
-        this.body.setPosition(0.0F, (float)(17 - var1), 2.0F);
-        this.leg1 = new ModelRenderer(0, 16);
-        this.leg1.addBox(-2.0F, 0.0F, -2.0F, 4, var1, 4, var2);
-        this.leg1.setPosition(-3.0F, (float)(24 - var1), 7.0F);
-        this.leg2 = new ModelRenderer(0, 16);
-        this.leg2.addBox(-2.0F, 0.0F, -2.0F, 4, var1, 4, var2);
-        this.leg2.setPosition(3.0F, (float)(24 - var1), 7.0F);
-        this.leg3 = new ModelRenderer(0, 16);
-        this.leg3.addBox(-2.0F, 0.0F, -2.0F, 4, var1, 4, var2);
-        this.leg3.setPosition(-3.0F, (float)(24 - var1), -5.0F);
-        this.leg4 = new ModelRenderer(0, 16);
-        this.leg4.addBox(-2.0F, 0.0F, -2.0F, 4, var1, 4, var2);
-        this.leg4.setPosition(3.0F, (float)(24 - var1), -5.0F);
-    }
-
-    public void render(float var1, float var2, float var3, float var4, float var5, float var6) {
-        this.setRotationAngles(var1, var2, var3, var4, var5, var6);
-        this.head.render(var6);
-        this.body.render(var6);
-        this.leg1.render(var6);
-        this.leg2.render(var6);
-        this.leg3.render(var6);
-        this.leg4.render(var6);
-    }
-
-    public void setRotationAngles(float var1, float var2, float var3, float var4, float var5, float var6) {
-        this.head.rotateAngleX = -(var5 / (180.0F / (float)Math.PI));
-        this.head.rotateAngleY = var4 / (180.0F / (float)Math.PI);
-        this.body.rotateAngleX = (float)Math.PI * 0.5F;
-        this.leg1.rotateAngleX = MathHelper.cos(var1 * 0.6662F) * 1.4F * var2;
-        this.leg2.rotateAngleX = MathHelper.cos(var1 * 0.6662F + (float)Math.PI) * 1.4F * var2;
-        this.leg3.rotateAngleX = MathHelper.cos(var1 * 0.6662F + (float)Math.PI) * 1.4F * var2;
-        this.leg4.rotateAngleX = MathHelper.cos(var1 * 0.6662F) * 1.4F * var2;
+    public ModelQuadraped(int i, float f) {
+        head = new ModelRenderer(0, 0);
+        head.addBox(-4F, -4F, -8F, 8, 8, 8, f);
+        head.setPosition(0.0F, 18 - i, -6F);
+        body = new ModelRenderer(28, 8);
+        body.addBox(-5F, -10F, -7F, 10, 16, 8, f);
+        body.setPosition(0.0F, 17 - i, 2.0F);
+        leg1 = new ModelRenderer(0, 16);
+        leg1.addBox(-2F, 0.0F, -2F, 4, i, 4, f);
+        leg1.setPosition(-3F, 24 - i, 7F);
+        leg2 = new ModelRenderer(0, 16);
+        leg2.addBox(-2F, 0.0F, -2F, 4, i, 4, f);
+        leg2.setPosition(3F, 24 - i, 7F);
+        leg3 = new ModelRenderer(0, 16);
+        leg3.addBox(-2F, 0.0F, -2F, 4, i, 4, f);
+        leg3.setPosition(-3F, 24 - i, -5F);
+        leg4 = new ModelRenderer(0, 16);
+        leg4.addBox(-2F, 0.0F, -2F, 4, i, 4, f);
+        leg4.setPosition(3F, 24 - i, -5F);
+    }
+
+    public void render(float f, float f1, float f2, float f3, float f4, float f5) {
+        setRotationAngles(f, f1, f2, f3, f4, f5);
+        head.render(f5);
+        body.render(f5);
+        leg1.render(f5);
+        leg2.render(f5);
+        leg3.render(f5);
+        leg4.render(f5);
+    }
+
+    public void setRotationAngles(float f, float f1, float f2, float f3, float f4, float f5) {
+        head.rotateAngleX = -(f4 / 57.29578F);
+        head.rotateAngleY = f3 / 57.29578F;
+        body.rotateAngleX = 1.570796F;
+        leg1.rotateAngleX = MathHelper.cos(f * 0.6662F) * 1.4F * f1;
+        leg2.rotateAngleX = MathHelper.cos(f * 0.6662F + 3.141593F) * 1.4F * f1;
+        leg3.rotateAngleX = MathHelper.cos(f * 0.6662F + 3.141593F) * 1.4F * f1;
+        leg4.rotateAngleX = MathHelper.cos(f * 0.6662F) * 1.4F * f1;
     }
 }
