--- net/minecraft/src/ModelBiped.java
+++ net/minecraft/src/ModelBiped.java
@@ -1,5 +1,8 @@
 package net.minecraft.src;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.world.player.model.PlayerModelRenderEvent;
+
 public class ModelBiped extends ModelBase {
     public ModelRenderer bipedHead;
     public ModelRenderer field_1285_b;
@@ -138,7 +141,7 @@
             this.bipedLeftLeg.offsetY = 12.0F;
             this.bipedHead.offsetY = 0.0F;
         }
-
+        Ref.getEventManager().handleEvent(new PlayerModelRenderEvent(this));
         this.bipedRightArm.rotateAngleZ += MathHelper.cos(var3 * 0.09F) * 0.05F + 0.05F;
         this.bipedLeftArm.rotateAngleZ -= MathHelper.cos(var3 * 0.09F) * 0.05F + 0.05F;
         this.bipedRightArm.rotateAngleX += MathHelper.sin(var3 * 0.067F) * 0.05F;
