--- net/minecraft/src/NBTTagLong.java
+++ net/minecraft/src/NBTTagLong.java
@@ -5,28 +5,29 @@
 import java.io.IOException;
 
 public class NBTTagLong extends NBTBase {
+
     public long longValue;
 
     public NBTTagLong() {
     }
 
-    public NBTTagLong(long var1) {
-        this.longValue = var1;
-    }
-
-    void writeTagContents(DataOutput var1) throws IOException {
-        var1.writeLong(this.longValue);
-    }
-
-    void readTagContents(DataInput var1) throws IOException {
-        this.longValue = var1.readLong();
+    public NBTTagLong(long l) {
+        longValue = l;
+    }
+
+    void writeTagContents(DataOutput dataoutput) throws IOException {
+        dataoutput.writeLong(longValue);
+    }
+
+    void readTagContents(DataInput datainput) throws IOException {
+        longValue = datainput.readLong();
     }
 
     public byte getType() {
-        return (byte)4;
+        return 4;
     }
 
     public String toString() {
-        return "" + this.longValue;
+        return String.valueOf(longValue);
     }
 }
