--- net/minecraft/src/NBTTagShort.java
+++ net/minecraft/src/NBTTagShort.java
@@ -5,28 +5,29 @@
 import java.io.IOException;
 
 public class NBTTagShort extends NBTBase {
+
     public short shortValue;
 
     public NBTTagShort() {
     }
 
-    public NBTTagShort(short var1) {
-        this.shortValue = var1;
-    }
-
-    void writeTagContents(DataOutput var1) throws IOException {
-        var1.writeShort(this.shortValue);
-    }
-
-    void readTagContents(DataInput var1) throws IOException {
-        this.shortValue = var1.readShort();
+    public NBTTagShort(short word0) {
+        shortValue = word0;
+    }
+
+    void writeTagContents(DataOutput dataoutput) throws IOException {
+        dataoutput.writeShort(shortValue);
+    }
+
+    void readTagContents(DataInput datainput) throws IOException {
+        shortValue = datainput.readShort();
     }
 
     public byte getType() {
-        return (byte)2;
+        return 2;
     }
 
     public String toString() {
-        return "" + this.shortValue;
+        return (new StringBuilder()).append("").append(shortValue).toString();
     }
 }
