--- net/minecraft/src/NetClientHandler.java
+++ net/minecraft/src/NetClientHandler.java
@@ -8,6 +8,10 @@
 import java.net.URL;
 import java.net.UnknownHostException;
 import java.util.Random;
+
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.chat.ChatReceivedEvent;
+import cc.noxiuam.titanic.event.impl.network.PacketTimeUpdateEvent;
 import net.minecraft.client.Minecraft;
 
 public class NetClientHandler extends NetHandler {
@@ -282,6 +286,9 @@
     }
 
     public void handleChat(Packet3Chat var1) {
+        ChatReceivedEvent event = new ChatReceivedEvent(var1.message);
+        Ref.getEventManager().handleEvent(event);
+
         this.mc.ingameGUI.addChatMessage(var1.message);
     }
 
@@ -354,6 +361,13 @@
     }
 
     public void handleUpdateTime(Packet4UpdateTime var1) {
+        PacketTimeUpdateEvent event = new PacketTimeUpdateEvent(var1);
+        Ref.getEventManager().handleEvent(event);
+
+        if (event.isCancelled()) {
+            return;
+        }
+
         this.mc.theWorld.setWorldTime(var1.time);
     }
 
