--- net/minecraft/src/NetworkManager.java
+++ net/minecraft/src/NetworkManager.java
@@ -9,6 +9,9 @@
 import java.util.Collections;
 import java.util.List;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.network.PacketReceivedEvent;
+
 public class NetworkManager {
     public static final Object threadSyncObject = new Object();
     public static int numReadThreads;
@@ -169,7 +172,8 @@
 
         while(!this.readPackets.isEmpty() && var1-- >= 0) {
             Packet var2 = (Packet)this.readPackets.remove(0);
-            var2.processPacket(this.netHandler);
+            Ref.getEventManager().handleEvent(new PacketReceivedEvent(var2));
+            var2.processPacket(netHandler);
         }
 
         if(this.isTerminating && this.readPackets.isEmpty()) {
