--- net/minecraft/src/NibbleArray.java
+++ net/minecraft/src/NibbleArray.java
@@ -1,36 +1,40 @@
 package net.minecraft.src;
 
 public class NibbleArray {
-    public final byte[] data;
-
-    public NibbleArray(int var1) {
-        this.data = new byte[var1 >> 1];
-    }
-
-    public NibbleArray(byte[] var1) {
-        this.data = var1;
-    }
-
-    public int getNibble(int var1, int var2, int var3) {
-        int var4 = var1 << 11 | var3 << 7 | var2;
-        int var5 = var4 >> 1;
-        int var6 = var4 & 1;
-        return var6 == 0 ? this.data[var5] & 15 : this.data[var5] >> 4 & 15;
-    }
-
-    public void setNibble(int var1, int var2, int var3, int var4) {
-        int var5 = var1 << 11 | var3 << 7 | var2;
-        int var6 = var5 >> 1;
-        int var7 = var5 & 1;
-        if(var7 == 0) {
-            this.data[var6] = (byte)(this.data[var6] & 240 | var4 & 15);
-        } else {
-            this.data[var6] = (byte)(this.data[var6] & 15 | (var4 & 15) << 4);
-        }
-
+
+    public final byte data[];
+
+    public NibbleArray(int i) {
+        data = new byte[i >> 1];
+    }
+
+    public NibbleArray(byte abyte0[]) {
+        data = abyte0;
+    }
+
+    public int getNibble(int i, int j, int k) {
+        int l = i << 11 | k << 7 | j;
+        int i1 = l >> 1;
+        int j1 = l & 1;
+        if (j1 == 0) {
+            return data[i1] & 0xf;
+        } else {
+            return data[i1] >> 4 & 0xf;
+        }
+    }
+
+    public void setNibble(int i, int j, int k, int l) {
+        int i1 = i << 11 | k << 7 | j;
+        int j1 = i1 >> 1;
+        int k1 = i1 & 1;
+        if (k1 == 0) {
+            data[j1] = (byte) (data[j1] & 0xf0 | l & 0xf);
+        } else {
+            data[j1] = (byte) (data[j1] & 0xf | (l & 0xf) << 4);
+        }
     }
 
     public boolean isValid() {
-        return this.data != null;
+        return data != null;
     }
 }
