--- net/minecraft/src/OpenGlCapsChecker.java
+++ net/minecraft/src/OpenGlCapsChecker.java
@@ -3,9 +3,14 @@
 import org.lwjgl.opengl.GLContext;
 
 public class OpenGlCapsChecker {
-    private static boolean tryCheckOcclusionCapable = false;
+
+    private static final boolean tryCheckOcclusionCapable = false;
+
+    public OpenGlCapsChecker() {
+    }
 
     public boolean checkARBOcclusion() {
         return tryCheckOcclusionCapable && GLContext.getCapabilities().GL_ARB_occlusion_query;
     }
+
 }
