--- net/minecraft/src/Packet100.java
+++ net/minecraft/src/Packet100.java
@@ -5,30 +5,34 @@
 import java.io.IOException;
 
 public class Packet100 extends Packet {
+
     public int field_20038_a;
     public int field_20037_b;
     public String field_20040_c;
     public int field_20039_d;
 
-    public void processPacket(NetHandler var1) {
-        var1.func_20087_a(this);
-    }
-
-    public void readPacketData(DataInputStream var1) throws IOException {
-        this.field_20038_a = var1.readByte();
-        this.field_20037_b = var1.readByte();
-        this.field_20040_c = var1.readUTF();
-        this.field_20039_d = var1.readByte();
-    }
-
-    public void writePacketData(DataOutputStream var1) throws IOException {
-        var1.writeByte(this.field_20038_a);
-        var1.writeByte(this.field_20037_b);
-        var1.writeUTF(this.field_20040_c);
-        var1.writeByte(this.field_20039_d);
+    public Packet100() {
+    }
+
+    public void processPacket(NetHandler nethandler) {
+        nethandler.func_20087_a(this);
+    }
+
+    public void readPacketData(DataInputStream datainputstream) throws IOException {
+        field_20038_a = datainputstream.readByte();
+        field_20037_b = datainputstream.readByte();
+        field_20040_c = datainputstream.readUTF();
+        field_20039_d = datainputstream.readByte();
+    }
+
+    public void writePacketData(DataOutputStream dataoutputstream) throws IOException {
+        dataoutputstream.writeByte(field_20038_a);
+        dataoutputstream.writeByte(field_20037_b);
+        dataoutputstream.writeUTF(field_20040_c);
+        dataoutputstream.writeByte(field_20039_d);
     }
 
     public int getPacketSize() {
-        return 3 + this.field_20040_c.length();
+        return 3 + field_20040_c.length();
     }
 }
