--- net/minecraft/src/Packet12PlayerLook.java
+++ net/minecraft/src/Packet12PlayerLook.java
@@ -5,27 +5,28 @@
 import java.io.IOException;
 
 public class Packet12PlayerLook extends Packet10Flying {
+
     public Packet12PlayerLook() {
-        this.rotating = true;
-    }
-
-    public Packet12PlayerLook(float var1, float var2, boolean var3) {
-        this.yaw = var1;
-        this.pitch = var2;
-        this.onGround = var3;
-        this.rotating = true;
-    }
-
-    public void readPacketData(DataInputStream var1) throws IOException {
-        this.yaw = var1.readFloat();
-        this.pitch = var1.readFloat();
-        super.readPacketData(var1);
-    }
-
-    public void writePacketData(DataOutputStream var1) throws IOException {
-        var1.writeFloat(this.yaw);
-        var1.writeFloat(this.pitch);
-        super.writePacketData(var1);
+        rotating = true;
+    }
+
+    public Packet12PlayerLook(float f, float f1, boolean flag) {
+        yaw = f;
+        pitch = f1;
+        onGround = flag;
+        rotating = true;
+    }
+
+    public void readPacketData(DataInputStream datainputstream) throws IOException {
+        yaw = datainputstream.readFloat();
+        pitch = datainputstream.readFloat();
+        super.readPacketData(datainputstream);
+    }
+
+    public void writePacketData(DataOutputStream dataoutputstream) throws IOException {
+        dataoutputstream.writeFloat(yaw);
+        dataoutputstream.writeFloat(pitch);
+        super.writePacketData(dataoutputstream);
     }
 
     public int getPacketSize() {
