--- net/minecraft/src/Packet9.java
+++ net/minecraft/src/Packet9.java
@@ -2,16 +2,21 @@
 
 import java.io.DataInputStream;
 import java.io.DataOutputStream;
+import java.io.IOException;
 
 public class Packet9 extends Packet {
-    public void processPacket(NetHandler var1) {
-        var1.func_9448_a(this);
-    }
-
-    public void readPacketData(DataInputStream var1) {
-    }
-
-    public void writePacketData(DataOutputStream var1) {
+
+    public Packet9() {
+    }
+
+    public void processPacket(NetHandler nethandler) {
+        nethandler.func_9448_a(this);
+    }
+
+    public void readPacketData(DataInputStream datainputstream) throws IOException {
+    }
+
+    public void writePacketData(DataOutputStream dataoutputstream) throws IOException {
     }
 
     public int getPacketSize() {
