--- net/minecraft/src/Packet39.java
+++ net/minecraft/src/Packet39.java
@@ -5,24 +5,28 @@
 import java.io.IOException;
 
 public class Packet39 extends Packet {
+
     public int field_6365_a;
     public int field_6364_b;
 
+    public Packet39() {
+    }
+
     public int getPacketSize() {
         return 8;
     }
 
-    public void readPacketData(DataInputStream var1) throws IOException {
-        this.field_6365_a = var1.readInt();
-        this.field_6364_b = var1.readInt();
-    }
-
-    public void writePacketData(DataOutputStream var1) throws IOException {
-        var1.writeInt(this.field_6365_a);
-        var1.writeInt(this.field_6364_b);
-    }
-
-    public void processPacket(NetHandler var1) {
-        var1.func_6497_a(this);
+    public void readPacketData(DataInputStream datainputstream) throws IOException {
+        field_6365_a = datainputstream.readInt();
+        field_6364_b = datainputstream.readInt();
+    }
+
+    public void writePacketData(DataOutputStream dataoutputstream) throws IOException {
+        dataoutputstream.writeInt(field_6365_a);
+        dataoutputstream.writeInt(field_6364_b);
+    }
+
+    public void processPacket(NetHandler nethandler) {
+        nethandler.func_6497_a(this);
     }
 }
