--- net/minecraft/src/Packet255KickDisconnect.java
+++ net/minecraft/src/Packet255KickDisconnect.java
@@ -5,28 +5,29 @@
 import java.io.IOException;
 
 public class Packet255KickDisconnect extends Packet {
+
     public String reason;
 
     public Packet255KickDisconnect() {
     }
 
-    public Packet255KickDisconnect(String var1) {
-        this.reason = var1;
-    }
-
-    public void readPacketData(DataInputStream var1) throws IOException {
-        this.reason = var1.readUTF();
-    }
-
-    public void writePacketData(DataOutputStream var1) throws IOException {
-        var1.writeUTF(this.reason);
-    }
-
-    public void processPacket(NetHandler var1) {
-        var1.handleKickDisconnect(this);
+    public Packet255KickDisconnect(String s) {
+        reason = s;
+    }
+
+    public void readPacketData(DataInputStream datainputstream) throws IOException {
+        reason = datainputstream.readUTF();
+    }
+
+    public void writePacketData(DataOutputStream dataoutputstream) throws IOException {
+        dataoutputstream.writeUTF(reason);
+    }
+
+    public void processPacket(NetHandler nethandler) {
+        nethandler.handleKickDisconnect(this);
     }
 
     public int getPacketSize() {
-        return this.reason.length();
+        return reason.length();
     }
 }
