--- net/minecraft/src/Packet2Handshake.java
+++ net/minecraft/src/Packet2Handshake.java
@@ -5,28 +5,29 @@
 import java.io.IOException;
 
 public class Packet2Handshake extends Packet {
+
     public String username;
 
     public Packet2Handshake() {
     }
 
-    public Packet2Handshake(String var1) {
-        this.username = var1;
-    }
-
-    public void readPacketData(DataInputStream var1) throws IOException {
-        this.username = var1.readUTF();
-    }
-
-    public void writePacketData(DataOutputStream var1) throws IOException {
-        var1.writeUTF(this.username);
-    }
-
-    public void processPacket(NetHandler var1) {
-        var1.handleHandshake(this);
+    public Packet2Handshake(String s) {
+        username = s;
+    }
+
+    public void readPacketData(DataInputStream datainputstream) throws IOException {
+        username = datainputstream.readUTF();
+    }
+
+    public void writePacketData(DataOutputStream dataoutputstream) throws IOException {
+        dataoutputstream.writeUTF(username);
+    }
+
+    public void processPacket(NetHandler nethandler) {
+        nethandler.handleHandshake(this);
     }
 
     public int getPacketSize() {
-        return 4 + this.username.length() + 4;
+        return 4 + username.length() + 4;
     }
 }
