--- net/minecraft/src/Packet10Flying.java
+++ net/minecraft/src/Packet10Flying.java
@@ -5,6 +5,7 @@
 import java.io.IOException;
 
 public class Packet10Flying extends Packet {
+
     public double xPosition;
     public double yPosition;
     public double zPosition;
@@ -14,24 +15,22 @@
     public boolean onGround;
     public boolean moving;
     public boolean rotating;
-
     public Packet10Flying() {
     }
-
-    public Packet10Flying(boolean var1) {
-        this.onGround = var1;
-    }
-
-    public void processPacket(NetHandler var1) {
-        var1.handleFlying(this);
-    }
-
-    public void readPacketData(DataInputStream var1) throws IOException {
-        this.onGround = var1.read() != 0;
-    }
-
-    public void writePacketData(DataOutputStream var1) throws IOException {
-        var1.write(this.onGround ? 1 : 0);
+    public Packet10Flying(boolean flag) {
+        onGround = flag;
+    }
+
+    public void processPacket(NetHandler nethandler) {
+        nethandler.handleFlying(this);
+    }
+
+    public void readPacketData(DataInputStream datainputstream) throws IOException {
+        onGround = datainputstream.read() != 0;
+    }
+
+    public void writePacketData(DataOutputStream dataoutputstream) throws IOException {
+        dataoutputstream.write(onGround ? 1 : 0);
     }
 
     public int getPacketSize() {
