--- net/minecraft/src/Packet30Entity.java
+++ net/minecraft/src/Packet30Entity.java
@@ -5,24 +5,28 @@
 import java.io.IOException;
 
 public class Packet30Entity extends Packet {
+
     public int entityId;
     public byte xPosition;
     public byte yPosition;
     public byte zPosition;
     public byte yaw;
     public byte pitch;
-    public boolean rotating = false;
-
-    public void readPacketData(DataInputStream var1) throws IOException {
-        this.entityId = var1.readInt();
-    }
-
-    public void writePacketData(DataOutputStream var1) throws IOException {
-        var1.writeInt(this.entityId);
-    }
-
-    public void processPacket(NetHandler var1) {
-        var1.handleEntity(this);
+    public boolean rotating;
+    public Packet30Entity() {
+        rotating = false;
+    }
+
+    public void readPacketData(DataInputStream datainputstream) throws IOException {
+        entityId = datainputstream.readInt();
+    }
+
+    public void writePacketData(DataOutputStream dataoutputstream) throws IOException {
+        dataoutputstream.writeInt(entityId);
+    }
+
+    public void processPacket(NetHandler nethandler) {
+        nethandler.handleEntity(this);
     }
 
     public int getPacketSize() {
