--- net/minecraft/src/Packet22Collect.java
+++ net/minecraft/src/Packet22Collect.java
@@ -5,21 +5,25 @@
 import java.io.IOException;
 
 public class Packet22Collect extends Packet {
+
     public int collectedEntityId;
     public int collectorEntityId;
 
-    public void readPacketData(DataInputStream var1) throws IOException {
-        this.collectedEntityId = var1.readInt();
-        this.collectorEntityId = var1.readInt();
-    }
-
-    public void writePacketData(DataOutputStream var1) throws IOException {
-        var1.writeInt(this.collectedEntityId);
-        var1.writeInt(this.collectorEntityId);
-    }
-
-    public void processPacket(NetHandler var1) {
-        var1.handleCollect(this);
+    public Packet22Collect() {
+    }
+
+    public void readPacketData(DataInputStream datainputstream) throws IOException {
+        collectedEntityId = datainputstream.readInt();
+        collectorEntityId = datainputstream.readInt();
+    }
+
+    public void writePacketData(DataOutputStream dataoutputstream) throws IOException {
+        dataoutputstream.writeInt(collectedEntityId);
+        dataoutputstream.writeInt(collectorEntityId);
+    }
+
+    public void processPacket(NetHandler nethandler) {
+        nethandler.handleCollect(this);
     }
 
     public int getPacketSize() {
