--- net/minecraft/src/Packet14BlockDig.java
+++ net/minecraft/src/Packet14BlockDig.java
@@ -5,6 +5,7 @@
 import java.io.IOException;
 
 public class Packet14BlockDig extends Packet {
+
     public int xPosition;
     public int yPosition;
     public int zPosition;
@@ -14,32 +15,32 @@
     public Packet14BlockDig() {
     }
 
-    public Packet14BlockDig(int var1, int var2, int var3, int var4, int var5) {
-        this.status = var1;
-        this.xPosition = var2;
-        this.yPosition = var3;
-        this.zPosition = var4;
-        this.face = var5;
-    }
-
-    public void readPacketData(DataInputStream var1) throws IOException {
-        this.status = var1.read();
-        this.xPosition = var1.readInt();
-        this.yPosition = var1.read();
-        this.zPosition = var1.readInt();
-        this.face = var1.read();
-    }
-
-    public void writePacketData(DataOutputStream var1) throws IOException {
-        var1.write(this.status);
-        var1.writeInt(this.xPosition);
-        var1.write(this.yPosition);
-        var1.writeInt(this.zPosition);
-        var1.write(this.face);
-    }
-
-    public void processPacket(NetHandler var1) {
-        var1.handleBlockDig(this);
+    public Packet14BlockDig(int i, int j, int k, int l, int i1) {
+        status = i;
+        xPosition = j;
+        yPosition = k;
+        zPosition = l;
+        face = i1;
+    }
+
+    public void readPacketData(DataInputStream datainputstream) throws IOException {
+        status = datainputstream.read();
+        xPosition = datainputstream.readInt();
+        yPosition = datainputstream.read();
+        zPosition = datainputstream.readInt();
+        face = datainputstream.read();
+    }
+
+    public void writePacketData(DataOutputStream dataoutputstream) throws IOException {
+        dataoutputstream.write(status);
+        dataoutputstream.writeInt(xPosition);
+        dataoutputstream.write(yPosition);
+        dataoutputstream.writeInt(zPosition);
+        dataoutputstream.write(face);
+    }
+
+    public void processPacket(NetHandler nethandler) {
+        nethandler.handleBlockDig(this);
     }
 
     public int getPacketSize() {
