--- net/minecraft/src/Packet4UpdateTime.java
+++ net/minecraft/src/Packet4UpdateTime.java
@@ -5,18 +5,22 @@
 import java.io.IOException;
 
 public class Packet4UpdateTime extends Packet {
+
     public long time;
 
-    public void readPacketData(DataInputStream var1) throws IOException {
-        this.time = var1.readLong();
-    }
-
-    public void writePacketData(DataOutputStream var1) throws IOException {
-        var1.writeLong(this.time);
-    }
-
-    public void processPacket(NetHandler var1) {
-        var1.handleUpdateTime(this);
+    public Packet4UpdateTime() {
+    }
+
+    public void readPacketData(DataInputStream datainputstream) throws IOException {
+        time = datainputstream.readLong();
+    }
+
+    public void writePacketData(DataOutputStream dataoutputstream) throws IOException {
+        dataoutputstream.writeLong(time);
+    }
+
+    public void processPacket(NetHandler nethandler) {
+        nethandler.handleUpdateTime(this);
     }
 
     public int getPacketSize() {
