--- net/minecraft/src/PlayerController.java
+++ net/minecraft/src/PlayerController.java
@@ -3,63 +3,63 @@
 import net.minecraft.client.Minecraft;
 
 public class PlayerController {
+
     protected final Minecraft mc;
-    public boolean field_1064_b = false;
-
-    public PlayerController(Minecraft var1) {
-        this.mc = var1;
-    }
-
-    public void func_717_a(World var1) {
-    }
-
-    public void clickBlock(int var1, int var2, int var3, int var4) {
-        this.sendBlockRemoved(var1, var2, var3, var4);
-    }
-
-    public boolean sendBlockRemoved(int var1, int var2, int var3, int var4) {
-        this.mc.effectRenderer.func_1186_a(var1, var2, var3);
-        World var5 = this.mc.theWorld;
-        Block var6 = Block.blocksList[var5.getBlockId(var1, var2, var3)];
-        int var7 = var5.getBlockMetadata(var1, var2, var3);
-        boolean var8 = var5.setBlockWithNotify(var1, var2, var3, 0);
-        if(var6 != null && var8) {
-            this.mc.sndManager.playSound(var6.stepSound.func_1146_a(), (float)var1 + 0.5F, (float)var2 + 0.5F, (float)var3 + 0.5F, (var6.stepSound.func_1147_b() + 1.0F) / 2.0F, var6.stepSound.func_1144_c() * 0.8F);
-            var6.onBlockDestroyedByPlayer(var5, var1, var2, var3, var7);
+    public boolean field_1064_b;
+
+    public PlayerController(Minecraft minecraft) {
+        field_1064_b = false;
+        mc = minecraft;
+    }
+
+    public void func_717_a(World world) {
+    }
+
+    public void clickBlock(int i, int j, int k, int l) {
+        sendBlockRemoved(i, j, k, l);
+    }
+
+    public boolean sendBlockRemoved(int i, int j, int k, int l) {
+        mc.effectRenderer.func_1186_a(i, j, k);
+        World world = mc.theWorld;
+        Block block = Block.blocksList[world.getBlockId(i, j, k)];
+        int i1 = world.getBlockMetadata(i, j, k);
+        boolean flag = world.setBlockWithNotify(i, j, k, 0);
+        if (block != null && flag) {
+            mc.sndManager.playSound(block.stepSound.func_1146_a(), (float) i + 0.5F, (float) j + 0.5F, (float) k + 0.5F, (block.stepSound.func_1147_b() + 1.0F) / 2.0F, block.stepSound.func_1144_c() * 0.8F);
+            block.onBlockDestroyedByPlayer(world, i, j, k, i1);
         }
-
-        return var8;
+        return flag;
     }
 
-    public void sendBlockRemoving(int var1, int var2, int var3, int var4) {
+    public void sendBlockRemoving(int i, int j, int k, int l) {
     }
 
     public void func_6468_a() {
     }
 
-    public void setPartialTime(float var1) {
+    public void setPartialTime(float f) {
     }
 
     public float getBlockReachDistance() {
-        return 5.0F;
+        return 5F;
     }
 
-    public boolean sendUseItem(EntityPlayer var1, World var2, ItemStack var3) {
-        int var4 = var3.stackSize;
-        ItemStack var5 = var3.useItemRightClick(var2, var1);
-        if(var5 != var3 || var5 != null && var5.stackSize != var4) {
-            var1.inventory.mainInventory[var1.inventory.currentItem] = var5;
-            if(var5.stackSize == 0) {
-                var1.inventory.mainInventory[var1.inventory.currentItem] = null;
+    public boolean sendUseItem(EntityPlayer entityplayer, World world, ItemStack itemstack) {
+        int i = itemstack.stackSize;
+        ItemStack itemstack1 = itemstack.useItemRightClick(world, entityplayer);
+        if (itemstack1 != itemstack || itemstack1 != null && itemstack1.stackSize != i) {
+            entityplayer.inventory.mainInventory[entityplayer.inventory.currentItem] = itemstack1;
+            if (itemstack1.stackSize == 0) {
+                entityplayer.inventory.mainInventory[entityplayer.inventory.currentItem] = null;
             }
-
             return true;
         } else {
             return false;
         }
     }
 
-    public void flipPlayer(EntityPlayer var1) {
+    public void flipPlayer(EntityPlayer entityplayer) {
     }
 
     public void updateController() {
@@ -69,32 +69,39 @@
         return true;
     }
 
-    public void func_6473_b(EntityPlayer var1) {
-    }
-
-    public boolean sendPlaceBlock(EntityPlayer var1, World var2, ItemStack var3, int var4, int var5, int var6, int var7) {
-        int var8 = var2.getBlockId(var4, var5, var6);
-        return var8 > 0 && Block.blocksList[var8].blockActivated(var2, var4, var5, var6, var1) ? true : (var3 == null ? false : var3.useItem(var1, var2, var4, var5, var6, var7));
-    }
-
-    public EntityPlayer func_4087_b(World var1) {
-        return new EntityPlayerSP(this.mc, var1, this.mc.session, var1.worldProvider.worldType);
-    }
-
-    public void func_6475_a(EntityPlayer var1, Entity var2) {
-        var1.useCurrentItemOnEntity(var2);
-    }
-
-    public void func_6472_b(EntityPlayer var1, Entity var2) {
-        var1.attackTargetEntityWithCurrentItem(var2);
-    }
-
-    public ItemStack func_20085_a(int var1, int var2, int var3, EntityPlayer var4) {
-        return var4.field_20068_h.func_20116_a(var2, var3, var4);
-    }
-
-    public void func_20086_a(int var1, EntityPlayer var2) {
-        var2.field_20068_h.onCraftGuiClosed(var2);
-        var2.field_20068_h = var2.field_20069_g;
+    public void func_6473_b(EntityPlayer entityplayer) {
+    }
+
+    public boolean sendPlaceBlock(EntityPlayer entityplayer, World world, ItemStack itemstack, int i, int j, int k, int l) {
+        int i1 = world.getBlockId(i, j, k);
+        if (i1 > 0 && Block.blocksList[i1].blockActivated(world, i, j, k, entityplayer)) {
+            return true;
+        }
+        if (itemstack == null) {
+            return false;
+        } else {
+            return itemstack.useItem(entityplayer, world, i, j, k, l);
+        }
+    }
+
+    public EntityPlayer func_4087_b(World world) {
+        return new EntityPlayerSP(mc, world, mc.session, world.worldProvider.worldType);
+    }
+
+    public void func_6475_a(EntityPlayer entityplayer, Entity entity) {
+        entityplayer.useCurrentItemOnEntity(entity);
+    }
+
+    public void func_6472_b(EntityPlayer entityplayer, Entity entity) {
+        entityplayer.attackTargetEntityWithCurrentItem(entity);
+    }
+
+    public ItemStack func_20085_a(int i, int j, int k, EntityPlayer entityplayer) {
+        return entityplayer.field_20068_h.func_20116_a(j, k, entityplayer);
+    }
+
+    public void func_20086_a(int i, EntityPlayer entityplayer) {
+        entityplayer.field_20068_h.onCraftGuiClosed(entityplayer);
+        entityplayer.field_20068_h = entityplayer.field_20069_g;
     }
 }
