--- net/minecraft/src/RecipeSorter.java
+++ net/minecraft/src/RecipeSorter.java
@@ -2,18 +2,23 @@
 
 import java.util.Comparator;
 
-class RecipeSorter implements Comparator {
-    final CraftingManager craftingManager;
-
-    RecipeSorter(CraftingManager var1) {
-        this.craftingManager = var1;
-    }
-
-    public int a(CraftingRecipe var1, CraftingRecipe var2) {
-        return var2.getRecipeSize() < var1.getRecipeSize() ? -1 : (var2.getRecipeSize() > var1.getRecipeSize() ? 1 : 0);
-    }
-
-    public int compare(Object var1, Object var2) {
-        return this.a((CraftingRecipe)var1, (CraftingRecipe)var2);
+class RecipeSorter
+        implements Comparator {
+
+    final CraftingManager craftingManager; /* synthetic field */
+
+    RecipeSorter(CraftingManager craftingmanager) {
+        craftingManager = craftingmanager;
+    }
+
+    public int compareRecipes(CraftingRecipe craftingrecipe, CraftingRecipe craftingrecipe1) {
+        if (craftingrecipe1.getRecipeSize() < craftingrecipe.getRecipeSize()) {
+            return -1;
+        }
+        return craftingrecipe1.getRecipeSize() <= craftingrecipe.getRecipeSize() ? 0 : 1;
+    }
+
+    public int compare(Object obj, Object obj1) {
+        return compareRecipes((CraftingRecipe) obj, (CraftingRecipe) obj1);
     }
 }
