--- net/minecraft/src/RecipesTools.java
+++ net/minecraft/src/RecipesTools.java
@@ -1,17 +1,46 @@
 package net.minecraft.src;
 
 public class RecipesTools {
-    private String[][] recipePatterns = new String[][]{{"XXX", " # ", " # "}, {"X", "#", "#"}, {"XX", "X#", " #"}, {"XX", " #", " #"}};
-    private Object[][] recipeItems = new Object[][]{{Block.planks, Block.cobblestone, Item.ingotIron, Item.diamond, Item.ingotGold}, {Item.pickaxeWood, Item.pickaxeStone, Item.pickaxeSteel, Item.pickaxeDiamond, Item.pickaxeGold}, {Item.shovelWood, Item.shovelStone, Item.shovelSteel, Item.shovelDiamond, Item.shovelGold}, {Item.axeWood, Item.axeStone, Item.axeSteel, Item.axeDiamond, Item.axeGold}, {Item.hoeWood, Item.hoeStone, Item.hoeSteel, Item.hoeDiamond, Item.hoeGold}};
-
-    public void addRecipes(CraftingManager var1) {
-        for(int var2 = 0; var2 < this.recipeItems[0].length; ++var2) {
-            Object var3 = this.recipeItems[0][var2];
-
-            for(int var4 = 0; var4 < this.recipeItems.length - 1; ++var4) {
-                Item var5 = (Item)this.recipeItems[var4 + 1][var2];
-                var1.addRecipe(new ItemStack(var5), new Object[]{this.recipePatterns[var4], Character.valueOf('#'), Item.stick, Character.valueOf('X'), var3});
+
+    private final String[][] recipePatterns = {
+            {
+                    "XXX", " # ", " # "
+            }, {
+            "X", "#", "#"
+    }, {
+            "XX", "X#", " #"
+    }, {
+            "XX", " #", " #"
+    }
+    };
+    private final Object[][] recipeItems;
+
+    public RecipesTools() {
+        recipeItems = (new Object[][]{
+                new Object[]{
+                        Block.planks, Block.cobblestone, Item.ingotIron, Item.diamond, Item.ingotGold
+                }, new Object[]{
+                Item.pickaxeWood, Item.pickaxeStone, Item.pickaxeSteel, Item.pickaxeDiamond, Item.pickaxeGold
+        }, new Object[]{
+                Item.shovelWood, Item.shovelStone, Item.shovelSteel, Item.shovelDiamond, Item.shovelGold
+        }, new Object[]{
+                Item.axeWood, Item.axeStone, Item.axeSteel, Item.axeDiamond, Item.axeGold
+        }, new Object[]{
+                Item.hoeWood, Item.hoeStone, Item.hoeSteel, Item.hoeDiamond, Item.hoeGold
+        }
+        });
+    }
+
+    public void addRecipes(CraftingManager craftingmanager) {
+        for (int i = 0; i < recipeItems[0].length; i++) {
+            Object obj = recipeItems[0][i];
+            for (int j = 0; j < recipeItems.length - 1; j++) {
+                Item item = (Item) recipeItems[j + 1][i];
+                craftingmanager.addRecipe(new ItemStack(item), new Object[]{
+                        recipePatterns[j], Character.valueOf('#'), Item.stick, Character.valueOf('X'), obj
+                });
             }
+
         }
 
     }
