--- net/minecraft/src/RenderGlobal.java
+++ net/minecraft/src/RenderGlobal.java
@@ -11,6 +11,9 @@
 import org.lwjgl.opengl.GL11;
 import org.lwjgl.opengl.GL15;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.world.CloudRenderEvent;
+
 public class RenderGlobal implements IWorldAccess {
     public List field_1458_a = new ArrayList();
     private World worldObj;
@@ -660,6 +663,13 @@
 
     public void func_4141_b(float var1) {
         if(!this.mc.theWorld.worldProvider.field_4220_c) {
+            CloudRenderEvent event = new CloudRenderEvent();
+            Ref.getEventManager().handleEvent(event);
+
+            if (event.isCancelled()) {
+                return;
+            }
+
             if(this.mc.gameSettings.fancyGraphics) {
                 this.func_6510_c(var1);
             } else {
