--- net/minecraft/src/RenderBlocks.java
+++ net/minecraft/src/RenderBlocks.java
@@ -2,6 +2,9 @@
 
 import org.lwjgl.opengl.GL11;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.world.block.BlockRenderEvent;
+
 public class RenderBlocks {
     private IBlockAccess blockAccess;
     private int overrideBlockTexture = -1;
@@ -22,6 +25,13 @@
     }
 
     public boolean renderBlockByRenderType(Block var1, int var2, int var3, int var4) {
+        BlockRenderEvent event = new BlockRenderEvent(var1);
+        Ref.getEventManager().handleEvent(event);
+
+        if (event.isCancelled()) {
+            return false;
+        }
+
         int var5 = var1.getRenderType();
         var1.setBlockBoundsBasedOnState(this.blockAccess, var2, var3, var4);
         return var5 == 0 ? this.renderStandardBlock(var1, var2, var3, var4) : (var5 == 4 ? this.renderBlockFluids(var1, var2, var3, var4) : (var5 == 13 ? this.renderBlockCactus(var1, var2, var3, var4) : (var5 == 1 ? this.renderBlockReed(var1, var2, var3, var4) : (var5 == 6 ? this.renderBlockCrops(var1, var2, var3, var4) : (var5 == 2 ? this.renderBlockTorch(var1, var2, var3, var4) : (var5 == 3 ? this.renderBlockFire(var1, var2, var3, var4) : (var5 == 5 ? this.renderBlockRedstoneWire(var1, var2, var3, var4) : (var5 == 8 ? this.renderBlockLadder(var1, var2, var3, var4) : (var5 == 7 ? this.renderBlockDoor(var1, var2, var3, var4) : (var5 == 9 ? this.renderBlockMinecartTrack(var1, var2, var3, var4) : (var5 == 10 ? this.renderBlockStairs(var1, var2, var3, var4) : (var5 == 11 ? this.renderBlockFence(var1, var2, var3, var4) : (var5 == 12 ? this.renderBlockLever(var1, var2, var3, var4) : false)))))))))))));
