--- net/minecraft/src/RenderList.java
+++ net/minecraft/src/RenderList.java
@@ -1,61 +1,71 @@
 package net.minecraft.src;
 
-import java.nio.IntBuffer;
 import org.lwjgl.opengl.GL11;
 
+import java.nio.IntBuffer;
+
 public class RenderList {
+
+    private final IntBuffer field_1236_g;
     private int field_1242_a;
     private int field_1241_b;
     private int field_1240_c;
     private float field_1239_d;
     private float field_1238_e;
     private float field_1237_f;
-    private IntBuffer field_1236_g = GLAllocation.createDirectIntBuffer(65536);
-    private boolean field_1235_h = false;
-    private boolean field_1234_i = false;
-
-    public void func_861_a(int var1, int var2, int var3, double var4, double var6, double var8) {
-        this.field_1235_h = true;
-        this.field_1236_g.clear();
-        this.field_1242_a = var1;
-        this.field_1241_b = var2;
-        this.field_1240_c = var3;
-        this.field_1239_d = (float)var4;
-        this.field_1238_e = (float)var6;
-        this.field_1237_f = (float)var8;
-    }
-
-    public boolean func_862_a(int var1, int var2, int var3) {
-        return !this.field_1235_h ? false : var1 == this.field_1242_a && var2 == this.field_1241_b && var3 == this.field_1240_c;
-    }
-
-    public void func_858_a(int var1) {
-        this.field_1236_g.put(var1);
-        if(this.field_1236_g.remaining() == 0) {
-            this.func_860_a();
-        }
-
+    private boolean field_1235_h;
+    private boolean field_1234_i;
+    public RenderList() {
+        field_1236_g = GLAllocation.createDirectIntBuffer(0x10000);
+        field_1235_h = false;
+        field_1234_i = false;
+    }
+
+    public void func_861_a(int i, int j, int k, double d, double d1,
+                           double d2) {
+        field_1235_h = true;
+        field_1236_g.clear();
+        field_1242_a = i;
+        field_1241_b = j;
+        field_1240_c = k;
+        field_1239_d = (float) d;
+        field_1238_e = (float) d1;
+        field_1237_f = (float) d2;
+    }
+
+    public boolean func_862_a(int i, int j, int k) {
+        if (!field_1235_h) {
+            return false;
+        } else {
+            return i == field_1242_a && j == field_1241_b && k == field_1240_c;
+        }
+    }
+
+    public void func_858_a(int i) {
+        field_1236_g.put(i);
+        if (field_1236_g.remaining() == 0) {
+            func_860_a();
+        }
     }
 
     public void func_860_a() {
-        if(this.field_1235_h) {
-            if(!this.field_1234_i) {
-                this.field_1236_g.flip();
-                this.field_1234_i = true;
-            }
-
-            if(this.field_1236_g.remaining() > 0) {
-                GL11.glPushMatrix();
-                GL11.glTranslatef((float)this.field_1242_a - this.field_1239_d, (float)this.field_1241_b - this.field_1238_e, (float)this.field_1240_c - this.field_1237_f);
-                GL11.glCallLists(this.field_1236_g);
-                GL11.glPopMatrix();
-            }
-
+        if (!field_1235_h) {
+            return;
+        }
+        if (!field_1234_i) {
+            field_1236_g.flip();
+            field_1234_i = true;
+        }
+        if (field_1236_g.remaining() > 0) {
+            GL11.glPushMatrix();
+            GL11.glTranslatef((float) field_1242_a - field_1239_d, (float) field_1241_b - field_1238_e, (float) field_1240_c - field_1237_f);
+            GL11.glCallLists(field_1236_g);
+            GL11.glPopMatrix();
         }
     }
 
     public void func_859_b() {
-        this.field_1235_h = false;
-        this.field_1234_i = false;
+        field_1235_h = false;
+        field_1234_i = false;
     }
 }
