--- net/minecraft/src/RenderLiving.java
+++ net/minecraft/src/RenderLiving.java
@@ -3,6 +3,9 @@
 import org.lwjgl.opengl.GL11;
 import org.lwjgl.opengl.GL12;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.world.player.model.SpecialModelRenderEvent;
+
 public class RenderLiving extends Render {
     protected ModelBase e;
     protected ModelBase renderPassModel;
@@ -21,6 +24,10 @@
         GL11.glDisable(GL11.GL_CULL_FACE);
         this.e.field_1244_k = this.func_167_c(var1, var9);
         this.e.field_1243_l = var1.ridingEntity != null || var1.field_9300_bu;
+
+        SpecialModelRenderEvent event = new SpecialModelRenderEvent(this.e, this.renderPassModel);
+        Ref.getEventManager().handleEvent(event);
+
         if(this.renderPassModel != null) {
             this.renderPassModel.field_1243_l = this.e.field_1243_l;
         }
