--- net/minecraft/src/RenderManager.java
+++ net/minecraft/src/RenderManager.java
@@ -5,6 +5,9 @@
 import java.util.Map;
 import org.lwjgl.opengl.GL11;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.world.entity.PreEntityRenderEvent;
+
 public class RenderManager {
     private Map entityRenderMap = new HashMap();
     public static RenderManager instance = new RenderManager();
@@ -98,6 +101,13 @@
     public void renderEntityWithPosYaw(Entity var1, double var2, double var4, double var6, float var8, float var9) {
         Render var10 = this.getEntityRenderObject(var1);
         if(var10 != null) {
+            PreEntityRenderEvent event = new PreEntityRenderEvent(var1);
+            Ref.getEventManager().handleEvent(event);
+
+            if (event.isCancelled()) {
+                return;
+            }
+
             var10.doRender(var1, var2, var4, var6, var8, var9);
             var10.doRenderShadowAndFire(var1, var2, var4, var6, var8, var9);
         }
