--- net/minecraft/src/RenderEntity.java
+++ net/minecraft/src/RenderEntity.java
@@ -3,9 +3,14 @@
 import org.lwjgl.opengl.GL11;
 
 public class RenderEntity extends Render {
-    public void doRender(Entity var1, double var2, double var4, double var6, float var8, float var9) {
+
+    public RenderEntity() {
+    }
+
+    public void doRender(Entity entity, double d, double d1, double d2,
+                         float f, float f1) {
         GL11.glPushMatrix();
-        renderOffsetAABB(var1.boundingBox, var2 - var1.lastTickPosX, var4 - var1.lastTickPosY, var6 - var1.lastTickPosZ);
+        renderOffsetAABB(entity.boundingBox, d - entity.lastTickPosX, d1 - entity.lastTickPosY, d2 - entity.lastTickPosZ);
         GL11.glPopMatrix();
     }
 }
