--- net/minecraft/src/RenderSheep.java
+++ net/minecraft/src/RenderSheep.java
@@ -1,17 +1,18 @@
 package net.minecraft.src;
 
 public class RenderSheep extends RenderLiving {
-    public RenderSheep(ModelBase var1, ModelBase var2, float var3) {
-        super(var1, var3);
-        this.setRenderPassModel(var2);
-    }
-
-    protected boolean a(EntitySheep var1, int var2) {
-        this.loadTexture("/mob/sheep_fur.png");
-        return var2 == 0 && !var1.sheared;
-    }
-
-    protected boolean shouldRenderPass(EntityLiving var1, int var2) {
-        return this.a((EntitySheep)var1, var2);
+
+    public RenderSheep(ModelBase modelbase, ModelBase modelbase1, float f) {
+        super(modelbase, f);
+        setRenderPassModel(modelbase1);
+    }
+
+    protected boolean func_176_a(EntitySheep entitysheep, int i) {
+        loadTexture("/mob/sheep_fur.png");
+        return i == 0 && !entitysheep.sheared;
+    }
+
+    protected boolean shouldRenderPass(EntityLiving entityliving, int i) {
+        return func_176_a((EntitySheep) entityliving, i);
     }
 }
