--- net/minecraft/src/RenderPlayer.java
+++ net/minecraft/src/RenderPlayer.java
@@ -2,6 +2,10 @@
 
 import org.lwjgl.opengl.GL11;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.world.player.NametagRenderEvent;
+import cc.noxiuam.titanic.event.impl.world.player.model.CapeRenderEvent;
+
 public class RenderPlayer extends RenderLiving {
     private ModelBiped field_209_f = (ModelBiped)this.e;
     private ModelBiped field_208_g = new ModelBiped(1.0F);
@@ -62,6 +66,13 @@
             GL11.glScalef(-var14, -var14, var14);
             String var18 = var1.field_771_i;
             GL11.glDisable(GL11.GL_LIGHTING);
+
+            NametagRenderEvent event = new NametagRenderEvent(var1, var14);
+            Ref.getEventManager().handleEvent(event);
+            if (event.isCancelled()) {
+                return;
+            }
+
             Tessellator var19;
             if(!var1.isSneaking()) {
                 GL11.glDepthMask(false);
@@ -180,6 +191,14 @@
 
             float var18 = var1.field_775_e + (var1.field_774_f - var1.field_775_e) * var2;
             var15 += MathHelper.sin((var1.prevDistanceWalkedModified + (var1.distanceWalkedModified - var1.prevDistanceWalkedModified) * var2) * 6.0F) * 32.0F * var18;
+
+            CapeRenderEvent event = new CapeRenderEvent(var1, var15);
+            Ref.getEventManager().handleEvent(event);
+
+            if (event.isCancelled()) {
+                var15 = event.value;
+            }
+
             GL11.glRotatef(6.0F + var16 / 2.0F + var15, 1.0F, 0.0F, 0.0F);
             GL11.glRotatef(var17 / 2.0F, 0.0F, 0.0F, 1.0F);
             GL11.glRotatef(-var17 / 2.0F, 0.0F, 1.0F, 0.0F);
