--- net/minecraft/src/RenderPig.java
+++ net/minecraft/src/RenderPig.java
@@ -1,17 +1,18 @@
 package net.minecraft.src;
 
 public class RenderPig extends RenderLiving {
-    public RenderPig(ModelBase var1, ModelBase var2, float var3) {
-        super(var1, var3);
-        this.setRenderPassModel(var2);
-    }
-
-    protected boolean a(EntityPig var1, int var2) {
-        this.loadTexture("/mob/saddle.png");
-        return var2 == 0 && var1.rideable;
-    }
-
-    protected boolean shouldRenderPass(EntityLiving var1, int var2) {
-        return this.a((EntityPig)var1, var2);
+
+    public RenderPig(ModelBase modelbase, ModelBase modelbase1, float f) {
+        super(modelbase, f);
+        setRenderPassModel(modelbase1);
+    }
+
+    protected boolean func_180_a(EntityPig entitypig, int i) {
+        loadTexture("/mob/saddle.png");
+        return i == 0 && entitypig.rideable;
+    }
+
+    protected boolean shouldRenderPass(EntityLiving entityliving, int i) {
+        return func_180_a((EntityPig) entityliving, i);
     }
 }
