--- net/minecraft/src/ScaledResolution.java
+++ net/minecraft/src/ScaledResolution.java
@@ -1,26 +1,25 @@
 package net.minecraft.src;
 
 public class ScaledResolution {
+
+    public int scaleFactor;
     private int scaledWidth;
     private int scaledHeight;
-    public int scaleFactor;
-
-    public ScaledResolution(int var1, int var2) {
-        this.scaledWidth = var1;
-        this.scaledHeight = var2;
-
-        for(this.scaleFactor = 1; this.scaledWidth / (this.scaleFactor + 1) >= 320 && this.scaledHeight / (this.scaleFactor + 1) >= 240; ++this.scaleFactor) {
+
+    public ScaledResolution(int i, int j) {
+        scaledWidth = i;
+        scaledHeight = j;
+        for (scaleFactor = 1; scaledWidth / (scaleFactor + 1) >= 320 && scaledHeight / (scaleFactor + 1) >= 240; scaleFactor++) {
         }
-
-        this.scaledWidth /= this.scaleFactor;
-        this.scaledHeight /= this.scaleFactor;
+        scaledWidth = scaledWidth / scaleFactor;
+        scaledHeight = scaledHeight / scaleFactor;
     }
 
     public int getScaledWidth() {
-        return this.scaledWidth;
+        return scaledWidth;
     }
 
     public int getScaledHeight() {
-        return this.scaledHeight;
+        return scaledHeight;
     }
 }
