--- net/minecraft/src/ScreenShotHelper.java
+++ net/minecraft/src/ScreenShotHelper.java
@@ -10,6 +10,9 @@
 import org.lwjgl.BufferUtils;
 import org.lwjgl.opengl.GL11;
 
+import cc.noxiuam.titanic.Ref;
+import cc.noxiuam.titanic.event.impl.SuccessfulScreenshotEvent;
+
 public class ScreenShotHelper {
     private static DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd_HH.mm.ss");
     private static ByteBuffer buffer;
@@ -53,6 +56,10 @@
                     BufferedImage var15 = new BufferedImage(var1, var2, 1);
                     var15.setRGB(0, 0, var1, var2, imageData, 0, var1);
                     ImageIO.write(var15, "png", var5);
+
+                    SuccessfulScreenshotEvent event = new SuccessfulScreenshotEvent(var5.getName());
+                    Ref.getEventManager().handleEvent(event);
+
                     return "Saved screenshot as " + var5.getName();
                 }
 
