--- net/minecraft/src/Session.java
+++ net/minecraft/src/Session.java
@@ -4,17 +4,11 @@
 import java.util.List;
 
 public class Session {
-    public static List registeredBlocksList = new ArrayList();
-    public String playerName;
-    public String field_6543_c;
-    public String field_6542_d;
 
-    public Session(String var1, String var2) {
-        this.playerName = var1;
-        this.field_6543_c = var2;
-    }
+    public static List registeredBlocksList;
 
     static {
+        registeredBlocksList = new ArrayList();
         registeredBlocksList.add(Block.stone);
         registeredBlocksList.add(Block.cobblestone);
         registeredBlocksList.add(Block.brick);
@@ -44,5 +38,14 @@
         registeredBlocksList.add(Block.tnt);
         registeredBlocksList.add(Block.obsidian);
         System.out.println(registeredBlocksList.size());
+    }
+
+    public String playerName;
+    public String field_6543_c;
+    public String field_6542_d;
+
+    public Session(String s, String s1) {
+        playerName = s;
+        field_6543_c = s1;
     }
 }
