--- net/minecraft/src/SoundPoolEntry.java
+++ net/minecraft/src/SoundPoolEntry.java
@@ -3,11 +3,11 @@
 import java.net.URL;
 
 public class SoundPoolEntry {
+
     public String soundName;
     public URL soundUrl;
-
-    public SoundPoolEntry(String var1, URL var2) {
-        this.soundName = var1;
-        this.soundUrl = var2;
+    public SoundPoolEntry(String s, URL url) {
+        soundName = s;
+        soundUrl = url;
     }
 }
