--- net/minecraft/src/Teleporter.java
+++ net/minecraft/src/Teleporter.java
@@ -3,258 +3,244 @@
 import java.util.Random;
 
 public class Teleporter {
-    private Random field_4232_a = new Random();
-
-    public void func_4107_a(World var1, Entity var2) {
-        if(!this.func_4106_b(var1, var2)) {
-            this.func_4108_c(var1, var2);
-            this.func_4106_b(var1, var2);
+
+    private final Random field_4232_a;
+
+    public Teleporter() {
+        field_4232_a = new Random();
+    }
+
+    public void func_4107_a(World world, Entity entity) {
+        if (func_4106_b(world, entity)) {
+        } else {
+            func_4108_c(world, entity);
+            func_4106_b(world, entity);
         }
     }
 
-    public boolean func_4106_b(World var1, Entity var2) {
-        short var3 = 128;
-        double var4 = -1.0D;
-        int var6 = 0;
-        int var7 = 0;
-        int var8 = 0;
-        int var9 = MathHelper.floor_double(var2.posX);
-        int var10 = MathHelper.floor_double(var2.posZ);
-
-        double var18;
-        for(int var11 = var9 - var3; var11 <= var9 + var3; ++var11) {
-            double var12 = (double)var11 + 0.5D - var2.posX;
-
-            for(int var14 = var10 - var3; var14 <= var10 + var3; ++var14) {
-                double var15 = (double)var14 + 0.5D - var2.posZ;
-
-                for(int var17 = 127; var17 >= 0; --var17) {
-                    if(var1.getBlockId(var11, var17, var14) == Block.portal.blockID) {
-                        while(var1.getBlockId(var11, var17 - 1, var14) == Block.portal.blockID) {
-                            --var17;
-                        }
-
-                        var18 = (double)var17 + 0.5D - var2.posY;
-                        double var20 = var12 * var12 + var18 * var18 + var15 * var15;
-                        if(var4 < 0.0D || var20 < var4) {
-                            var4 = var20;
-                            var6 = var11;
-                            var7 = var17;
-                            var8 = var14;
-                        }
+    public boolean func_4106_b(World world, Entity entity) {
+        char c = '\200';
+        double d = -1D;
+        int i = 0;
+        int j = 0;
+        int k = 0;
+        int l = MathHelper.floor_double(entity.posX);
+        int i1 = MathHelper.floor_double(entity.posZ);
+        for (int j1 = l - c; j1 <= l + c; j1++) {
+            double d1 = ((double) j1 + 0.5D) - entity.posX;
+            for (int j2 = i1 - c; j2 <= i1 + c; j2++) {
+                double d3 = ((double) j2 + 0.5D) - entity.posZ;
+                for (int k2 = 127; k2 >= 0; k2--) {
+                    if (world.getBlockId(j1, k2, j2) != Block.portal.blockID) {
+                        continue;
+                    }
+                    for (; world.getBlockId(j1, k2 - 1, j2) == Block.portal.blockID; k2--) {
+                    }
+                    double d5 = ((double) k2 + 0.5D) - entity.posY;
+                    double d7 = d1 * d1 + d5 * d5 + d3 * d3;
+                    if (d < 0.0D || d7 < d) {
+                        d = d7;
+                        i = j1;
+                        j = k2;
+                        k = j2;
                     }
                 }
+
             }
+
         }
 
-        if(var4 >= 0.0D) {
-            double var22 = (double)var6 + 0.5D;
-            double var16 = (double)var7 + 0.5D;
-            var18 = (double)var8 + 0.5D;
-            if(var1.getBlockId(var6 - 1, var7, var8) == Block.portal.blockID) {
-                var22 -= 0.5D;
-            }
-
-            if(var1.getBlockId(var6 + 1, var7, var8) == Block.portal.blockID) {
-                var22 += 0.5D;
-            }
-
-            if(var1.getBlockId(var6, var7, var8 - 1) == Block.portal.blockID) {
-                var18 -= 0.5D;
-            }
-
-            if(var1.getBlockId(var6, var7, var8 + 1) == Block.portal.blockID) {
-                var18 += 0.5D;
-            }
-
-            System.out.println("Teleporting to " + var22 + ", " + var16 + ", " + var18);
-            var2.setLocationAndAngles(var22, var16, var18, var2.rotationYaw, 0.0F);
-            var2.motionX = var2.motionY = var2.motionZ = 0.0D;
+        if (d >= 0.0D) {
+            int k1 = i;
+            int l1 = j;
+            int i2 = k;
+            double d2 = (double) k1 + 0.5D;
+            double d4 = (double) l1 + 0.5D;
+            double d6 = (double) i2 + 0.5D;
+            if (world.getBlockId(k1 - 1, l1, i2) == Block.portal.blockID) {
+                d2 -= 0.5D;
+            }
+            if (world.getBlockId(k1 + 1, l1, i2) == Block.portal.blockID) {
+                d2 += 0.5D;
+            }
+            if (world.getBlockId(k1, l1, i2 - 1) == Block.portal.blockID) {
+                d6 -= 0.5D;
+            }
+            if (world.getBlockId(k1, l1, i2 + 1) == Block.portal.blockID) {
+                d6 += 0.5D;
+            }
+            System.out.println("Teleporting to " + d2 + ", " + d4 + ", " + d6);
+            entity.setLocationAndAngles(d2, d4, d6, entity.rotationYaw, 0.0F);
+            entity.motionX = entity.motionY = entity.motionZ = 0.0D;
             return true;
         } else {
             return false;
         }
     }
 
-    public boolean func_4108_c(World var1, Entity var2) {
-        byte var3 = 16;
-        double var4 = -1.0D;
-        int var6 = MathHelper.floor_double(var2.posX);
-        int var7 = MathHelper.floor_double(var2.posY);
-        int var8 = MathHelper.floor_double(var2.posZ);
-        int var9 = var6;
-        int var10 = var7;
-        int var11 = var8;
-        int var12 = 0;
-        int var13 = this.field_4232_a.nextInt(4);
-
-        int var14;
-        double var15;
-        int var17;
-        double var18;
-        int var20;
-        int var21;
-        int var22;
-        int var23;
-        int var24;
-        int var25;
-        int var26;
-        int var27;
-        int var28;
-        double var32;
-        double var33;
-        for(var14 = var6 - var3; var14 <= var6 + var3; ++var14) {
-            var15 = (double)var14 + 0.5D - var2.posX;
-
-            for(var17 = var8 - var3; var17 <= var8 + var3; ++var17) {
-                var18 = (double)var17 + 0.5D - var2.posZ;
-
-                label293:
-                for(var20 = 127; var20 >= 0; --var20) {
-                    if(var1.func_20084_d(var14, var20, var17)) {
-                        while(var20 > 0 && var1.func_20084_d(var14, var20 - 1, var17)) {
-                            --var20;
-                        }
-
-                        for(var21 = var13; var21 < var13 + 4; ++var21) {
-                            var22 = var21 % 2;
-                            var23 = 1 - var22;
-                            if(var21 % 4 >= 2) {
-                                var22 = -var22;
-                                var23 = -var23;
-                            }
-
-                            for(var24 = 0; var24 < 3; ++var24) {
-                                for(var25 = 0; var25 < 4; ++var25) {
-                                    for(var26 = -1; var26 < 4; ++var26) {
-                                        var27 = var14 + (var25 - 1) * var22 + var24 * var23;
-                                        var28 = var20 + var26;
-                                        int var29 = var17 + (var25 - 1) * var23 - var24 * var22;
-                                        if(var26 < 0 && !var1.getBlockMaterial(var27, var28, var29).func_878_a() || var26 >= 0 && !var1.func_20084_d(var27, var28, var29)) {
-                                            continue label293;
-                                        }
-                                    }
-                                }
-                            }
-
-                            var32 = (double)var20 + 0.5D - var2.posY;
-                            var33 = var15 * var15 + var32 * var32 + var18 * var18;
-                            if(var4 < 0.0D || var33 < var4) {
-                                var4 = var33;
-                                var9 = var14;
-                                var10 = var20;
-                                var11 = var17;
-                                var12 = var21 % 4;
-                            }
-                        }
-                    }
-                }
-            }
-        }
-
-        if(var4 < 0.0D) {
-            for(var14 = var6 - var3; var14 <= var6 + var3; ++var14) {
-                var15 = (double)var14 + 0.5D - var2.posX;
-
-                for(var17 = var8 - var3; var17 <= var8 + var3; ++var17) {
-                    var18 = (double)var17 + 0.5D - var2.posZ;
-
-                    label231:
-                    for(var20 = 127; var20 >= 0; --var20) {
-                        if(var1.func_20084_d(var14, var20, var17)) {
-                            while(var1.func_20084_d(var14, var20 - 1, var17)) {
-                                --var20;
-                            }
-
-                            for(var21 = var13; var21 < var13 + 2; ++var21) {
-                                var22 = var21 % 2;
-                                var23 = 1 - var22;
-
-                                for(var24 = 0; var24 < 4; ++var24) {
-                                    for(var25 = -1; var25 < 4; ++var25) {
-                                        var26 = var14 + (var24 - 1) * var22;
-                                        var27 = var20 + var25;
-                                        var28 = var17 + (var24 - 1) * var23;
-                                        if(var25 < 0 && !var1.getBlockMaterial(var26, var27, var28).func_878_a() || var25 >= 0 && !var1.func_20084_d(var26, var27, var28)) {
-                                            continue label231;
-                                        }
-                                    }
-                                }
-
-                                var32 = (double)var20 + 0.5D - var2.posY;
-                                var33 = var15 * var15 + var32 * var32 + var18 * var18;
-                                if(var4 < 0.0D || var33 < var4) {
-                                    var4 = var33;
-                                    var9 = var14;
-                                    var10 = var20;
-                                    var11 = var17;
-                                    var12 = var21 % 2;
-                                }
-                            }
-                        }
-                    }
-                }
-            }
-        }
-
-        int var30 = var9;
-        int var16 = var10;
-        var17 = var11;
-        int var31 = var12 % 2;
-        int var19 = 1 - var31;
-        if(var12 % 4 >= 2) {
-            var31 = -var31;
-            var19 = -var19;
-        }
-
-        boolean var34;
-        if(var4 < 0.0D) {
-            if(var10 < 70) {
-                var10 = 70;
-            }
-
-            if(var10 > 118) {
-                var10 = 118;
-            }
-
-            var16 = var10;
-
-            for(var20 = -1; var20 <= 1; ++var20) {
-                for(var21 = 1; var21 < 3; ++var21) {
-                    for(var22 = -1; var22 < 3; ++var22) {
-                        var23 = var30 + (var21 - 1) * var31 + var20 * var19;
-                        var24 = var16 + var22;
-                        var25 = var17 + (var21 - 1) * var19 - var20 * var31;
-                        var34 = var22 < 0;
-                        var1.setBlockWithNotify(var23, var24, var25, var34 ? Block.obsidian.blockID : 0);
-                    }
-                }
-            }
-        }
-
-        for(var20 = 0; var20 < 4; ++var20) {
-            var1.field_1043_h = true;
-
-            for(var21 = 0; var21 < 4; ++var21) {
-                for(var22 = -1; var22 < 4; ++var22) {
-                    var23 = var30 + (var21 - 1) * var31;
-                    var24 = var16 + var22;
-                    var25 = var17 + (var21 - 1) * var19;
-                    var34 = var21 == 0 || var21 == 3 || var22 == -1 || var22 == 3;
-                    var1.setBlockWithNotify(var23, var24, var25, var34 ? Block.obsidian.blockID : Block.portal.blockID);
-                }
-            }
-
-            var1.field_1043_h = false;
-
-            for(var21 = 0; var21 < 4; ++var21) {
-                for(var22 = -1; var22 < 4; ++var22) {
-                    var23 = var30 + (var21 - 1) * var31;
-                    var24 = var16 + var22;
-                    var25 = var17 + (var21 - 1) * var19;
-                    var1.notifyBlocksOfNeighborChange(var23, var24, var25, var1.getBlockId(var23, var24, var25));
-                }
-            }
+    public boolean func_4108_c(World world, Entity entity) {
+        byte byte0 = 16;
+        double d = -1D;
+        int i = MathHelper.floor_double(entity.posX);
+        int j = MathHelper.floor_double(entity.posY);
+        int k = MathHelper.floor_double(entity.posZ);
+        int l = i;
+        int i1 = j;
+        int j1 = k;
+        int k1 = 0;
+        int l1 = field_4232_a.nextInt(4);
+        for (int i2 = i - byte0; i2 <= i + byte0; i2++) {
+            double d1 = ((double) i2 + 0.5D) - entity.posX;
+            for (int j3 = k - byte0; j3 <= k + byte0; j3++) {
+                double d3 = ((double) j3 + 0.5D) - entity.posZ;
+                for (int k4 = 127; k4 >= 0; k4--) {
+                    if (!world.func_20084_d(i2, k4, j3)) {
+                        continue;
+                    }
+                    for (; k4 > 0 && world.func_20084_d(i2, k4 - 1, j3); k4--) {
+                    }
+                    label0:
+                    for (int k5 = l1; k5 < l1 + 4; k5++) {
+                        int l6 = k5 % 2;
+                        int i8 = 1 - l6;
+                        if (k5 % 4 >= 2) {
+                            l6 = -l6;
+                            i8 = -i8;
+                        }
+                        for (int j9 = 0; j9 < 3; j9++) {
+                            for (int k10 = 0; k10 < 4; k10++) {
+                                for (int l11 = -1; l11 < 4; l11++) {
+                                    int j12 = i2 + (k10 - 1) * l6 + j9 * i8;
+                                    int l12 = k4 + l11;
+                                    int j13 = (j3 + (k10 - 1) * i8) - j9 * l6;
+                                    if (l11 < 0 && !world.getBlockMaterial(j12, l12, j13).func_878_a() || l11 >= 0 && !world.func_20084_d(j12, l12, j13)) {
+                                        break label0;
+                                    }
+                                }
+
+                            }
+
+                        }
+
+                        double d5 = ((double) k4 + 0.5D) - entity.posY;
+                        double d7 = d1 * d1 + d5 * d5 + d3 * d3;
+                        if (d < 0.0D || d7 < d) {
+                            d = d7;
+                            l = i2;
+                            i1 = k4;
+                            j1 = j3;
+                            k1 = k5 % 4;
+                        }
+                    }
+
+                }
+
+            }
+
+        }
+
+        if (d < 0.0D) {
+            for (int j2 = i - byte0; j2 <= i + byte0; j2++) {
+                double d2 = ((double) j2 + 0.5D) - entity.posX;
+                for (int k3 = k - byte0; k3 <= k + byte0; k3++) {
+                    double d4 = ((double) k3 + 0.5D) - entity.posZ;
+                    for (int l4 = 127; l4 >= 0; l4--) {
+                        if (!world.func_20084_d(j2, l4, k3)) {
+                            continue;
+                        }
+                        for (; world.func_20084_d(j2, l4 - 1, k3); l4--) {
+                        }
+                        label1:
+                        for (int l5 = l1; l5 < l1 + 2; l5++) {
+                            int i7 = l5 % 2;
+                            int j8 = 1 - i7;
+                            for (int k9 = 0; k9 < 4; k9++) {
+                                for (int l10 = -1; l10 < 4; l10++) {
+                                    int i12 = j2 + (k9 - 1) * i7;
+                                    int k12 = l4 + l10;
+                                    int i13 = k3 + (k9 - 1) * j8;
+                                    if (l10 < 0 && !world.getBlockMaterial(i12, k12, i13).func_878_a() || l10 >= 0 && !world.func_20084_d(i12, k12, i13)) {
+                                        break label1;
+                                    }
+                                }
+
+                            }
+
+                            double d6 = ((double) l4 + 0.5D) - entity.posY;
+                            double d8 = d2 * d2 + d6 * d6 + d4 * d4;
+                            if (d < 0.0D || d8 < d) {
+                                d = d8;
+                                l = j2;
+                                i1 = l4;
+                                j1 = k3;
+                                k1 = l5 % 2;
+                            }
+                        }
+
+                    }
+
+                }
+
+            }
+
+        }
+        int k2 = k1;
+        int l2 = l;
+        int i3 = i1;
+        int l3 = j1;
+        int i4 = k2 % 2;
+        int j4 = 1 - i4;
+        if (k2 % 4 >= 2) {
+            i4 = -i4;
+            j4 = -j4;
+        }
+        if (d < 0.0D) {
+            if (i1 < 70) {
+                i1 = 70;
+            }
+            if (i1 > 118) {
+                i1 = 118;
+            }
+            i3 = i1;
+            for (int i5 = -1; i5 <= 1; i5++) {
+                for (int i6 = 1; i6 < 3; i6++) {
+                    for (int j7 = -1; j7 < 3; j7++) {
+                        int k8 = l2 + (i6 - 1) * i4 + i5 * j4;
+                        int l9 = i3 + j7;
+                        int i11 = (l3 + (i6 - 1) * j4) - i5 * i4;
+                        boolean flag = j7 < 0;
+                        world.setBlockWithNotify(k8, l9, i11, flag ? Block.obsidian.blockID : 0);
+                    }
+
+                }
+
+            }
+
+        }
+        for (int j5 = 0; j5 < 4; j5++) {
+            world.field_1043_h = true;
+            for (int j6 = 0; j6 < 4; j6++) {
+                for (int k7 = -1; k7 < 4; k7++) {
+                    int l8 = l2 + (j6 - 1) * i4;
+                    int i10 = i3 + k7;
+                    int j11 = l3 + (j6 - 1) * j4;
+                    boolean flag1 = j6 == 0 || j6 == 3 || k7 == -1 || k7 == 3;
+                    world.setBlockWithNotify(l8, i10, j11, flag1 ? Block.obsidian.blockID : Block.portal.blockID);
+                }
+
+            }
+
+            world.field_1043_h = false;
+            for (int k6 = 0; k6 < 4; k6++) {
+                for (int l7 = -1; l7 < 4; l7++) {
+                    int i9 = l2 + (k6 - 1) * i4;
+                    int j10 = i3 + l7;
+                    int k11 = l3 + (k6 - 1) * j4;
+                    world.notifyBlocksOfNeighborChange(i9, j10, k11, world.getBlockId(i9, j10, k11));
+                }
+
+            }
+
         }
 
         return true;
