--- net/minecraft/src/TexturePackDefault.java
+++ net/minecraft/src/TexturePackDefault.java
@@ -1,44 +1,42 @@
 package net.minecraft.src;
 
-import java.awt.image.BufferedImage;
-import java.io.IOException;
-import javax.imageio.ImageIO;
 import net.minecraft.client.Minecraft;
 import org.lwjgl.opengl.GL11;
 
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+
 public class TexturePackDefault extends TexturePackBase {
-    private int texturePackName = -1;
+
+    private int texturePackName;
     private BufferedImage texturePackThumbnail;
 
     public TexturePackDefault() {
-        this.texturePackFileName = "Default";
-        this.firstDescriptionLine = "The default look of Minecraft";
-
+        texturePackName = -1;
+        texturePackFileName = "Default";
+        firstDescriptionLine = "The default look of Minecraft";
         try {
-            this.texturePackThumbnail = ImageIO.read(TexturePackDefault.class.getResource("/pack.png"));
-        } catch (IOException var2) {
-            var2.printStackTrace();
-        }
-
-    }
-
-    public void func_6484_b(Minecraft var1) {
-        if(this.texturePackThumbnail != null) {
-            var1.renderEngine.deleteTexture(this.texturePackName);
-        }
-
-    }
-
-    public void func_6483_c(Minecraft var1) {
-        if(this.texturePackThumbnail != null && this.texturePackName < 0) {
-            this.texturePackName = var1.renderEngine.allocateAndSetupTexture(this.texturePackThumbnail);
-        }
-
-        if(this.texturePackThumbnail != null) {
-            var1.renderEngine.bindTexture(this.texturePackName);
+            texturePackThumbnail = ImageIO.read((TexturePackDefault.class).getResource("/pack.png"));
+        } catch (IOException ioexception) {
+            ioexception.printStackTrace();
+        }
+    }
+
+    public void func_6484_b(Minecraft minecraft) {
+        if (texturePackThumbnail != null) {
+            minecraft.renderEngine.deleteTexture(texturePackName);
+        }
+    }
+
+    public void func_6483_c(Minecraft minecraft) {
+        if (texturePackThumbnail != null && texturePackName < 0) {
+            texturePackName = minecraft.renderEngine.allocateAndSetupTexture(texturePackThumbnail);
+        }
+        if (texturePackThumbnail != null) {
+            minecraft.renderEngine.bindTexture(texturePackName);
         } else {
-            GL11.glBindTexture(GL11.GL_TEXTURE_2D, var1.renderEngine.getTexture("/gui/unknown_pack.png"));
+            GL11.glBindTexture(3553 /*GL_TEXTURE_2D*/, minecraft.renderEngine.getTexture("/gui/unknown_pack.png"));
         }
-
     }
 }
