--- net/minecraft/src/ThreadRunIsoClient.java
+++ net/minecraft/src/ThreadRunIsoClient.java
@@ -1,21 +1,20 @@
 package net.minecraft.src;
 
 class ThreadRunIsoClient extends Thread {
-    final CanvasIsomPreview isoCanvas;
-
-    ThreadRunIsoClient(CanvasIsomPreview var1) {
-        this.isoCanvas = var1;
+
+    final CanvasIsomPreview isoCanvas; /* synthetic field */
+
+    ThreadRunIsoClient(CanvasIsomPreview canvasisompreview) {
+        isoCanvas = canvasisompreview;
     }
 
     public void run() {
-        while(CanvasIsomPreview.isRunning(this.isoCanvas)) {
-            this.isoCanvas.func_1265_d();
-
+        while (CanvasIsomPreview.isRunning(isoCanvas)) {
+            isoCanvas.func_1265_d();
             try {
                 Thread.sleep(1L);
-            } catch (Exception var2) {
+            } catch (Exception exception) {
             }
         }
-
     }
 }
