--- net/minecraft/src/ThreadSleepForever.java
+++ net/minecraft/src/ThreadSleepForever.java
@@ -3,22 +3,22 @@
 import net.minecraft.client.Minecraft;
 
 public class ThreadSleepForever extends Thread {
-    final Minecraft mc;
-
-    public ThreadSleepForever(Minecraft var1, String var2) {
-        super(var2);
-        this.mc = var1;
-        this.setDaemon(true);
-        this.start();
+
+    final Minecraft mc; /* synthetic field */
+
+    public ThreadSleepForever(Minecraft minecraft, String s) {
+        super(s);
+        mc = minecraft;
+        setDaemon(true);
+        start();
     }
 
     public void run() {
-        while(this.mc.running) {
+        while (mc.running) {
             try {
-                Thread.sleep(2147483647L);
-            } catch (InterruptedException var2) {
+                Thread.sleep(0x7fffffffL);
+            } catch (InterruptedException interruptedexception) {
             }
         }
-
     }
 }
