--- net/minecraft/src/WorldGenerator.java
+++ net/minecraft/src/WorldGenerator.java
@@ -3,8 +3,12 @@
 import java.util.Random;
 
 public abstract class WorldGenerator {
-    public abstract boolean generate(World var1, Random var2, int var3, int var4, int var5);
-
-    public void func_517_a(double var1, double var3, double var5) {
+
+    public WorldGenerator() {
+    }
+
+    public abstract boolean generate(World world, Random random, int i, int j, int k);
+
+    public void func_517_a(double d, double d1, double d2) {
     }
 }
