--- net/minecraft/src/WorldGenDungeons.java
+++ net/minecraft/src/WorldGenDungeons.java
@@ -3,114 +3,161 @@
 import java.util.Random;
 
 public class WorldGenDungeons extends WorldGenerator {
-    public boolean generate(World var1, Random var2, int var3, int var4, int var5) {
-        byte var6 = 3;
-        int var7 = var2.nextInt(2) + 2;
-        int var8 = var2.nextInt(2) + 2;
-        int var9 = 0;
-
-        int var10;
-        int var11;
-        int var12;
-        for(var10 = var3 - var7 - 1; var10 <= var3 + var7 + 1; ++var10) {
-            for(var11 = var4 - 1; var11 <= var4 + var6 + 1; ++var11) {
-                for(var12 = var5 - var8 - 1; var12 <= var5 + var8 + 1; ++var12) {
-                    Material var13 = var1.getBlockMaterial(var10, var11, var12);
-                    if(var11 == var4 - 1 && !var13.func_878_a()) {
-                        return false;
-                    }
-
-                    if(var11 == var4 + var6 + 1 && !var13.func_878_a()) {
-                        return false;
-                    }
-
-                    if((var10 == var3 - var7 - 1 || var10 == var3 + var7 + 1 || var12 == var5 - var8 - 1 || var12 == var5 + var8 + 1) && var11 == var4 && var1.func_20084_d(var10, var11, var12) && var1.func_20084_d(var10, var11 + 1, var12)) {
-                        ++var9;
+
+    public WorldGenDungeons() {
+    }
+
+    public boolean generate(World world, Random random, int i, int j, int k) {
+        byte byte0 = 3;
+        int l = random.nextInt(2) + 2;
+        int i1 = random.nextInt(2) + 2;
+        int j1 = 0;
+        for (int k1 = i - l - 1; k1 <= i + l + 1; k1++) {
+            for (int j2 = j - 1; j2 <= j + byte0 + 1; j2++) {
+                for (int i3 = k - i1 - 1; i3 <= k + i1 + 1; i3++) {
+                    Material material = world.getBlockMaterial(k1, j2, i3);
+                    if (j2 == j - 1 && !material.func_878_a()) {
+                        return false;
+                    }
+                    if (j2 == j + byte0 + 1 && !material.func_878_a()) {
+                        return false;
+                    }
+                    if ((k1 == i - l - 1 || k1 == i + l + 1 || i3 == k - i1 - 1 || i3 == k + i1 + 1) && j2 == j && world.func_20084_d(k1, j2, i3) && world.func_20084_d(k1, j2 + 1, i3)) {
+                        j1++;
                     }
                 }
+
             }
+
         }
 
-        if(var9 >= 1 && var9 <= 5) {
-            for(var10 = var3 - var7 - 1; var10 <= var3 + var7 + 1; ++var10) {
-                for(var11 = var4 + var6; var11 >= var4 - 1; --var11) {
-                    for(var12 = var5 - var8 - 1; var12 <= var5 + var8 + 1; ++var12) {
-                        if(var10 != var3 - var7 - 1 && var11 != var4 - 1 && var12 != var5 - var8 - 1 && var10 != var3 + var7 + 1 && var11 != var4 + var6 + 1 && var12 != var5 + var8 + 1) {
-                            var1.setBlockWithNotify(var10, var11, var12, 0);
-                        } else if(var11 >= 0 && !var1.getBlockMaterial(var10, var11 - 1, var12).func_878_a()) {
-                            var1.setBlockWithNotify(var10, var11, var12, 0);
-                        } else if(var1.getBlockMaterial(var10, var11, var12).func_878_a()) {
-                            if(var11 == var4 - 1 && var2.nextInt(4) != 0) {
-                                var1.setBlockWithNotify(var10, var11, var12, Block.cobblestoneMossy.blockID);
-                            } else {
-                                var1.setBlockWithNotify(var10, var11, var12, Block.cobblestone.blockID);
-                            }
-                        }
-                    }
-                }
-            }
-
-            label110:
-            for(var10 = 0; var10 < 2; ++var10) {
-                for(var11 = 0; var11 < 3; ++var11) {
-                    var12 = var3 + var2.nextInt(var7 * 2 + 1) - var7;
-                    int var14 = var5 + var2.nextInt(var8 * 2 + 1) - var8;
-                    if(var1.func_20084_d(var12, var4, var14)) {
-                        int var15 = 0;
-                        if(var1.getBlockMaterial(var12 - 1, var4, var14).func_878_a()) {
-                            ++var15;
-                        }
-
-                        if(var1.getBlockMaterial(var12 + 1, var4, var14).func_878_a()) {
-                            ++var15;
-                        }
-
-                        if(var1.getBlockMaterial(var12, var4, var14 - 1).func_878_a()) {
-                            ++var15;
-                        }
-
-                        if(var1.getBlockMaterial(var12, var4, var14 + 1).func_878_a()) {
-                            ++var15;
-                        }
-
-                        if(var15 == 1) {
-                            var1.setBlockWithNotify(var12, var4, var14, Block.crate.blockID);
-                            TileEntityChest var16 = (TileEntityChest)var1.getBlockTileEntity(var12, var4, var14);
-                            int var17 = 0;
-
-                            while(true) {
-                                if(var17 >= 8) {
-                                    continue label110;
-                                }
-
-                                ItemStack var18 = this.pickCheckLootItem(var2);
-                                if(var18 != null) {
-                                    var16.setInventorySlotContents(var2.nextInt(var16.getSizeInventory()), var18);
-                                }
-
-                                ++var17;
-                            }
-                        }
-                    }
-                }
-            }
-
-            var1.setBlockWithNotify(var3, var4, var5, Block.mobSpawner.blockID);
-            TileEntityMobSpawner var19 = (TileEntityMobSpawner)var1.getBlockTileEntity(var3, var4, var5);
-            var19.entityID = this.pickMobSpawner(var2);
-            return true;
-        } else {
+        if (j1 < 1 || j1 > 5) {
             return false;
         }
-    }
-
-    private ItemStack pickCheckLootItem(Random var1) {
-        int var2 = var1.nextInt(11);
-        return var2 == 0 ? new ItemStack(Item.saddle) : (var2 == 1 ? new ItemStack(Item.ingotIron, var1.nextInt(4) + 1) : (var2 == 2 ? new ItemStack(Item.bread) : (var2 == 3 ? new ItemStack(Item.wheat, var1.nextInt(4) + 1) : (var2 == 4 ? new ItemStack(Item.gunpowder, var1.nextInt(4) + 1) : (var2 == 5 ? new ItemStack(Item.silk, var1.nextInt(4) + 1) : (var2 == 6 ? new ItemStack(Item.bucketEmpty) : (var2 == 7 && var1.nextInt(100) == 0 ? new ItemStack(Item.appleGold) : (var2 == 8 && var1.nextInt(2) == 0 ? new ItemStack(Item.redstone, var1.nextInt(4) + 1) : (var2 == 9 && var1.nextInt(10) == 0 ? new ItemStack(Item.itemsList[Item.record13.shiftedIndex + var1.nextInt(2)]) : null)))))))));
-    }
-
-    private String pickMobSpawner(Random var1) {
-        int var2 = var1.nextInt(4);
-        return var2 == 0 ? "Skeleton" : (var2 == 1 ? "Zombie" : (var2 == 2 ? "Zombie" : (var2 == 3 ? "Spider" : "")));
+        for (int l1 = i - l - 1; l1 <= i + l + 1; l1++) {
+            for (int k2 = j + byte0; k2 >= j - 1; k2--) {
+                for (int j3 = k - i1 - 1; j3 <= k + i1 + 1; j3++) {
+                    if (l1 == i - l - 1 || k2 == j - 1 || j3 == k - i1 - 1 || l1 == i + l + 1 || k2 == j + byte0 + 1 || j3 == k + i1 + 1) {
+                        if (k2 >= 0 && !world.getBlockMaterial(l1, k2 - 1, j3).func_878_a()) {
+                            world.setBlockWithNotify(l1, k2, j3, 0);
+                            continue;
+                        }
+                        if (!world.getBlockMaterial(l1, k2, j3).func_878_a()) {
+                            continue;
+                        }
+                        if (k2 == j - 1 && random.nextInt(4) != 0) {
+                            world.setBlockWithNotify(l1, k2, j3, Block.cobblestoneMossy.blockID);
+                        } else {
+                            world.setBlockWithNotify(l1, k2, j3, Block.cobblestone.blockID);
+                        }
+                    } else {
+                        world.setBlockWithNotify(l1, k2, j3, 0);
+                    }
+                }
+
+            }
+
+        }
+
+        for (int i2 = 0; i2 < 2; i2++) {
+            label0:
+            for (int l2 = 0; l2 < 3; l2++) {
+                int k3 = (i + random.nextInt(l * 2 + 1)) - l;
+                int l3 = j;
+                int i4 = (k + random.nextInt(i1 * 2 + 1)) - i1;
+                if (!world.func_20084_d(k3, l3, i4)) {
+                    continue;
+                }
+                int j4 = 0;
+                if (world.getBlockMaterial(k3 - 1, l3, i4).func_878_a()) {
+                    j4++;
+                }
+                if (world.getBlockMaterial(k3 + 1, l3, i4).func_878_a()) {
+                    j4++;
+                }
+                if (world.getBlockMaterial(k3, l3, i4 - 1).func_878_a()) {
+                    j4++;
+                }
+                if (world.getBlockMaterial(k3, l3, i4 + 1).func_878_a()) {
+                    j4++;
+                }
+                if (j4 != 1) {
+                    continue;
+                }
+                world.setBlockWithNotify(k3, l3, i4, Block.crate.blockID);
+                TileEntityChest tileentitychest = (TileEntityChest) world.getBlockTileEntity(k3, l3, i4);
+                int k4 = 0;
+                do {
+                    if (k4 >= 8) {
+                        break label0;
+                    }
+                    ItemStack itemstack = pickCheckLootItem(random);
+                    if (itemstack != null) {
+                        tileentitychest.setInventorySlotContents(random.nextInt(tileentitychest.getSizeInventory()), itemstack);
+                    }
+                    k4++;
+                } while (true);
+            }
+
+        }
+
+        world.setBlockWithNotify(i, j, k, Block.mobSpawner.blockID);
+        TileEntityMobSpawner tileentitymobspawner = (TileEntityMobSpawner) world.getBlockTileEntity(i, j, k);
+        tileentitymobspawner.entityID = pickMobSpawner(random);
+        return true;
+    }
+
+    private ItemStack pickCheckLootItem(Random random) {
+        int i = random.nextInt(11);
+        if (i == 0) {
+            return new ItemStack(Item.saddle);
+        }
+        if (i == 1) {
+            return new ItemStack(Item.ingotIron, random.nextInt(4) + 1);
+        }
+        if (i == 2) {
+            return new ItemStack(Item.bread);
+        }
+        if (i == 3) {
+            return new ItemStack(Item.wheat, random.nextInt(4) + 1);
+        }
+        if (i == 4) {
+            return new ItemStack(Item.gunpowder, random.nextInt(4) + 1);
+        }
+        if (i == 5) {
+            return new ItemStack(Item.silk, random.nextInt(4) + 1);
+        }
+        if (i == 6) {
+            return new ItemStack(Item.bucketEmpty);
+        }
+        if (i == 7 && random.nextInt(100) == 0) {
+            return new ItemStack(Item.appleGold);
+        }
+        if (i == 8 && random.nextInt(2) == 0) {
+            return new ItemStack(Item.redstone, random.nextInt(4) + 1);
+        }
+        if (i == 9 && random.nextInt(10) == 0) {
+            return new ItemStack(Item.itemsList[Item.record13.shiftedIndex + random.nextInt(2)]);
+        } else {
+            return null;
+        }
+    }
+
+    private String pickMobSpawner(Random random) {
+        int i = random.nextInt(4);
+        if (i == 0) {
+            return "Skeleton";
+        }
+        if (i == 1) {
+            return "Zombie";
+        }
+        if (i == 2) {
+            return "Zombie";
+        }
+        if (i == 3) {
+            return "Spider";
+        } else {
+            return "";
+        }
     }
 }
